// klaviyo-api/dist/model/abTestActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbTestActionData = void 0;
class AbTestActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = AbTestActionData.StatusEnum.Draft;
        /**
        * The status of the A/B test action experiment.
        */
        this['experimentStatus'] = AbTestActionData.ExperimentStatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return AbTestActionData.attributeTypeMap;
    }
}
exports.AbTestActionData = AbTestActionData;
AbTestActionData.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "AbTestActionData.StatusEnum"
    },
    {
        "name": "experimentStatus",
        "baseName": "experiment_status",
        "type": "AbTestActionData.ExperimentStatusEnum"
    },
    {
        "name": "mainAction",
        "baseName": "main_action",
        "type": "AbTestActionDataMainAction"
    },
    {
        "name": "currentExperiment",
        "baseName": "current_experiment",
        "type": "AbTestActionDataCurrentExperiment"
    }
];
(function (AbTestActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = AbTestActionData.StatusEnum || (AbTestActionData.StatusEnum = {}));
    let ExperimentStatusEnum;
    (function (ExperimentStatusEnum) {
        ExperimentStatusEnum[ExperimentStatusEnum["Completed"] = 'completed'] = "Completed";
        ExperimentStatusEnum[ExperimentStatusEnum["Draft"] = 'draft'] = "Draft";
        ExperimentStatusEnum[ExperimentStatusEnum["Live"] = 'live'] = "Live";
    })(ExperimentStatusEnum = AbTestActionData.ExperimentStatusEnum || (AbTestActionData.ExperimentStatusEnum = {}));
})(AbTestActionData = exports.AbTestActionData || (exports.AbTestActionData = {}));
