// klaviyo-api/dist/model/abTestActionDataCurrentExperiment.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbTestActionDataCurrentExperiment = void 0;
class AbTestActionDataCurrentExperiment {
    static getAttributeTypeMap() {
        return AbTestActionDataCurrentExperiment.attributeTypeMap;
    }
}
exports.AbTestActionDataCurrentExperiment = AbTestActionDataCurrentExperiment;
AbTestActionDataCurrentExperiment.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "variations",
        "baseName": "variations",
        "type": "Array<AbTestActionDataMainAction>"
    },
    {
        "name": "allocations",
        "baseName": "allocations",
        "type": "object"
    },
    {
        "name": "started",
        "baseName": "started",
        "type": "Date"
    },
    {
        "name": "winnerMetric",
        "baseName": "winner_metric",
        "type": "AbTestActionDataCurrentExperiment.WinnerMetricEnum"
    },
    {
        "name": "automaticWinnerSelectionSettings",
        "baseName": "automatic_winner_selection_settings",
        "type": "AutomaticWinnerSelectionSettings"
    }
];
(function (AbTestActionDataCurrentExperiment) {
    let WinnerMetricEnum;
    (function (WinnerMetricEnum) {
        WinnerMetricEnum[WinnerMetricEnum["Submission"] = 'submission'] = "Submission";
        WinnerMetricEnum[WinnerMetricEnum["UniqueClicks"] = 'unique-clicks'] = "UniqueClicks";
        WinnerMetricEnum[WinnerMetricEnum["UniqueOpens"] = 'unique-opens'] = "UniqueOpens";
        WinnerMetricEnum[WinnerMetricEnum["UniquePlacedOrders"] = 'unique-placed-orders'] = "UniquePlacedOrders";
    })(WinnerMetricEnum = AbTestActionDataCurrentExperiment.WinnerMetricEnum || (AbTestActionDataCurrentExperiment.WinnerMetricEnum = {}));
})(AbTestActionDataCurrentExperiment = exports.AbTestActionDataCurrentExperiment || (exports.AbTestActionDataCurrentExperiment = {}));
