// klaviyo-api/dist/model/anniversaryDateFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnniversaryDateFilter = void 0;
class AnniversaryDateFilter {
    static getAttributeTypeMap() {
        return AnniversaryDateFilter.attributeTypeMap;
    }
}
exports.AnniversaryDateFilter = AnniversaryDateFilter;
AnniversaryDateFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "AnniversaryDateFilter.OperatorEnum"
    }
];
(function (AnniversaryDateFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Anniversary"] = 'anniversary'] = "Anniversary";
        OperatorEnum[OperatorEnum["AnniversaryMonth"] = 'anniversary-month'] = "AnniversaryMonth";
    })(OperatorEnum = AnniversaryDateFilter.OperatorEnum || (AnniversaryDateFilter.OperatorEnum = {}));
})(AnniversaryDateFilter = exports.AnniversaryDateFilter || (exports.AnniversaryDateFilter = {}));
