// klaviyo-api/dist/model/blockDisplayOptions.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockDisplayOptions = void 0;
class BlockDisplayOptions {
    static getAttributeTypeMap() {
        return BlockDisplayOptions.attributeTypeMap;
    }
}
exports.BlockDisplayOptions = BlockDisplayOptions;
BlockDisplayOptions.attributeTypeMap = [
    {
        "name": "showOn",
        "baseName": "show_on",
        "type": "BlockDisplayOptions.ShowOnEnum"
    },
    {
        "name": "visibleCheck",
        "baseName": "visible_check",
        "type": "string"
    },
    {
        "name": "contentRepeat",
        "baseName": "content_repeat",
        "type": "ContentRepeat"
    }
];
(function (BlockDisplayOptions) {
    let ShowOnEnum;
    (function (ShowOnEnum) {
        ShowOnEnum[ShowOnEnum["All"] = 'all'] = "All";
        ShowOnEnum[ShowOnEnum["Desktop"] = 'desktop'] = "Desktop";
        ShowOnEnum[ShowOnEnum["Mobile"] = 'mobile'] = "Mobile";
    })(ShowOnEnum = BlockDisplayOptions.ShowOnEnum || (BlockDisplayOptions.ShowOnEnum = {}));
})(BlockDisplayOptions = exports.BlockDisplayOptions || (exports.BlockDisplayOptions = {}));
