// klaviyo-api/dist/model/bulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes = void 0;
class BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes {
    constructor() {
        /**
        * The total number of operations that have been completed by the job.
        */
        this['completedCount'] = 0;
        /**
        * The total number of profiles that have been skipped as part of the job.
        */
        this['skippedCount'] = 0;
    }
    static getAttributeTypeMap() {
        return BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes = BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes;
BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "createdAt",
        "baseName": "created_at",
        "type": "Date"
    },
    {
        "name": "totalCount",
        "baseName": "total_count",
        "type": "number"
    },
    {
        "name": "completedCount",
        "baseName": "completed_count",
        "type": "number"
    },
    {
        "name": "completedAt",
        "baseName": "completed_at",
        "type": "Date"
    },
    {
        "name": "skippedCount",
        "baseName": "skipped_count",
        "type": "number"
    }
];
(function (BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Cancelled"] = 'cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["Complete"] = 'complete'] = "Complete";
        StatusEnum[StatusEnum["Processing"] = 'processing'] = "Processing";
        StatusEnum[StatusEnum["Queued"] = 'queued'] = "Queued";
    })(StatusEnum = BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.StatusEnum || (BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes.StatusEnum = {}));
})(BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes = exports.BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes || (exports.BulkProfileSuppressionsCreateJobResponseObjectResourceAttributes = {}));
