// klaviyo-api/dist/model/campaignResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignResponseObjectResourceAttributes = void 0;
class CampaignResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return CampaignResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.CampaignResponseObjectResourceAttributes = CampaignResponseObjectResourceAttributes;
CampaignResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CampaignResponseObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean"
    },
    {
        "name": "audiences",
        "baseName": "audiences",
        "type": "Audiences"
    },
    {
        "name": "sendOptions",
        "baseName": "send_options",
        "type": "CampaignResponseObjectResourceAttributesSendOptions"
    },
    {
        "name": "trackingOptions",
        "baseName": "tracking_options",
        "type": "CampaignResponseObjectResourceAttributesTrackingOptions"
    },
    {
        "name": "sendStrategy",
        "baseName": "send_strategy",
        "type": "CampaignResponseObjectResourceAttributesSendStrategy"
    },
    {
        "name": "createdAt",
        "baseName": "created_at",
        "type": "Date"
    },
    {
        "name": "scheduledAt",
        "baseName": "scheduled_at",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updated_at",
        "type": "Date"
    },
    {
        "name": "sendTime",
        "baseName": "send_time",
        "type": "Date"
    }
];
(function (CampaignResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AddingRecipients"] = 'Adding Recipients'] = "AddingRecipients";
        StatusEnum[StatusEnum["Cancelled"] = 'Cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["CancelledAccountDisabled"] = 'Cancelled: Account Disabled'] = "CancelledAccountDisabled";
        StatusEnum[StatusEnum["CancelledInternalError"] = 'Cancelled: Internal Error'] = "CancelledInternalError";
        StatusEnum[StatusEnum["CancelledNoRecipients"] = 'Cancelled: No Recipients'] = "CancelledNoRecipients";
        StatusEnum[StatusEnum["CancelledSmartSending"] = 'Cancelled: Smart Sending'] = "CancelledSmartSending";
        StatusEnum[StatusEnum["Draft"] = 'Draft'] = "Draft";
        StatusEnum[StatusEnum["PreparingToSchedule"] = 'Preparing to schedule'] = "PreparingToSchedule";
        StatusEnum[StatusEnum["PreparingToSend"] = 'Preparing to send'] = "PreparingToSend";
        StatusEnum[StatusEnum["QueuedWithoutRecipients"] = 'Queued without Recipients'] = "QueuedWithoutRecipients";
        StatusEnum[StatusEnum["Scheduled"] = 'Scheduled'] = "Scheduled";
        StatusEnum[StatusEnum["Sending"] = 'Sending'] = "Sending";
        StatusEnum[StatusEnum["SendingSegments"] = 'Sending Segments'] = "SendingSegments";
        StatusEnum[StatusEnum["Sent"] = 'Sent'] = "Sent";
        StatusEnum[StatusEnum["Unknown"] = 'Unknown'] = "Unknown";
        StatusEnum[StatusEnum["VariationsSent"] = 'Variations Sent'] = "VariationsSent";
    })(StatusEnum = CampaignResponseObjectResourceAttributes.StatusEnum || (CampaignResponseObjectResourceAttributes.StatusEnum = {}));
})(CampaignResponseObjectResourceAttributes = exports.CampaignResponseObjectResourceAttributes || (exports.CampaignResponseObjectResourceAttributes = {}));
