// klaviyo-api/dist/model/campaignRecipientEstimationJobResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignRecipientEstimationJobResponseObjectResourceAttributes = void 0;
class CampaignRecipientEstimationJobResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return CampaignRecipientEstimationJobResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.CampaignRecipientEstimationJobResponseObjectResourceAttributes = CampaignRecipientEstimationJobResponseObjectResourceAttributes;
CampaignRecipientEstimationJobResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "CampaignRecipientEstimationJobResponseObjectResourceAttributes.StatusEnum"
    }
];
(function (CampaignRecipientEstimationJobResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Cancelled"] = 'cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["Complete"] = 'complete'] = "Complete";
        StatusEnum[StatusEnum["Processing"] = 'processing'] = "Processing";
        StatusEnum[StatusEnum["Queued"] = 'queued'] = "Queued";
    })(StatusEnum = CampaignRecipientEstimationJobResponseObjectResourceAttributes.StatusEnum || (CampaignRecipientEstimationJobResponseObjectResourceAttributes.StatusEnum = {}));
})(CampaignRecipientEstimationJobResponseObjectResourceAttributes = exports.CampaignRecipientEstimationJobResponseObjectResourceAttributes || (exports.CampaignRecipientEstimationJobResponseObjectResourceAttributes = {}));
