// klaviyo-api/dist/model/campaignSendJobPartialUpdateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignSendJobPartialUpdateQueryResourceObjectAttributes = void 0;
class CampaignSendJobPartialUpdateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return CampaignSendJobPartialUpdateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.CampaignSendJobPartialUpdateQueryResourceObjectAttributes = CampaignSendJobPartialUpdateQueryResourceObjectAttributes;
CampaignSendJobPartialUpdateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "CampaignSendJobPartialUpdateQueryResourceObjectAttributes.ActionEnum"
    }
];
(function (CampaignSendJobPartialUpdateQueryResourceObjectAttributes) {
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum[ActionEnum["Cancel"] = 'cancel'] = "Cancel";
        ActionEnum[ActionEnum["Revert"] = 'revert'] = "Revert";
    })(ActionEnum = CampaignSendJobPartialUpdateQueryResourceObjectAttributes.ActionEnum || (CampaignSendJobPartialUpdateQueryResourceObjectAttributes.ActionEnum = {}));
})(CampaignSendJobPartialUpdateQueryResourceObjectAttributes = exports.CampaignSendJobPartialUpdateQueryResourceObjectAttributes || (exports.CampaignSendJobPartialUpdateQueryResourceObjectAttributes = {}));
