// klaviyo-api/dist/model/campaignSendJobResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignSendJobResponseObjectResourceAttributes = void 0;
class CampaignSendJobResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return CampaignSendJobResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.CampaignSendJobResponseObjectResourceAttributes = CampaignSendJobResponseObjectResourceAttributes;
CampaignSendJobResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "CampaignSendJobResponseObjectResourceAttributes.StatusEnum"
    }
];
(function (CampaignSendJobResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Cancelled"] = 'cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["Complete"] = 'complete'] = "Complete";
        StatusEnum[StatusEnum["Processing"] = 'processing'] = "Processing";
        StatusEnum[StatusEnum["Queued"] = 'queued'] = "Queued";
    })(StatusEnum = CampaignSendJobResponseObjectResourceAttributes.StatusEnum || (CampaignSendJobResponseObjectResourceAttributes.StatusEnum = {}));
})(CampaignSendJobResponseObjectResourceAttributes = exports.CampaignSendJobResponseObjectResourceAttributes || (exports.CampaignSendJobResponseObjectResourceAttributes = {}));
