// klaviyo-api/dist/model/catalogVariantUpdateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogVariantUpdateQueryResourceObjectAttributes = void 0;
class CatalogVariantUpdateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return CatalogVariantUpdateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.CatalogVariantUpdateQueryResourceObjectAttributes = CatalogVariantUpdateQueryResourceObjectAttributes;
CatalogVariantUpdateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "sku",
        "baseName": "sku",
        "type": "string"
    },
    {
        "name": "inventoryPolicy",
        "baseName": "inventory_policy",
        "type": "CatalogVariantUpdateQueryResourceObjectAttributes.InventoryPolicyEnum"
    },
    {
        "name": "inventoryQuantity",
        "baseName": "inventory_quantity",
        "type": "number"
    },
    {
        "name": "price",
        "baseName": "price",
        "type": "number"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "imageFullUrl",
        "baseName": "image_full_url",
        "type": "string"
    },
    {
        "name": "imageThumbnailUrl",
        "baseName": "image_thumbnail_url",
        "type": "string"
    },
    {
        "name": "images",
        "baseName": "images",
        "type": "Array<string>"
    },
    {
        "name": "customMetadata",
        "baseName": "custom_metadata",
        "type": "object"
    },
    {
        "name": "published",
        "baseName": "published",
        "type": "boolean"
    }
];
(function (CatalogVariantUpdateQueryResourceObjectAttributes) {
    let InventoryPolicyEnum;
    (function (InventoryPolicyEnum) {
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_0"] = 0] = "NUMBER_0";
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_1"] = 1] = "NUMBER_1";
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_2"] = 2] = "NUMBER_2";
    })(InventoryPolicyEnum = CatalogVariantUpdateQueryResourceObjectAttributes.InventoryPolicyEnum || (CatalogVariantUpdateQueryResourceObjectAttributes.InventoryPolicyEnum = {}));
})(CatalogVariantUpdateQueryResourceObjectAttributes = exports.CatalogVariantUpdateQueryResourceObjectAttributes || (exports.CatalogVariantUpdateQueryResourceObjectAttributes = {}));
