// klaviyo-api/dist/model/catalogCategoryCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogCategoryCreateQueryResourceObjectAttributes = void 0;
class CatalogCategoryCreateQueryResourceObjectAttributes {
    constructor() {
        /**
        * The integration type. Currently only \"$custom\" is supported.
        */
        this['integrationType'] = CatalogCategoryCreateQueryResourceObjectAttributes.IntegrationTypeEnum.Custom;
        /**
        * The type of catalog. Currently only \"$default\" is supported.
        */
        this['catalogType'] = '$default';
    }
    static getAttributeTypeMap() {
        return CatalogCategoryCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.CatalogCategoryCreateQueryResourceObjectAttributes = CatalogCategoryCreateQueryResourceObjectAttributes;
CatalogCategoryCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "externalId",
        "baseName": "external_id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "integrationType",
        "baseName": "integration_type",
        "type": "CatalogCategoryCreateQueryResourceObjectAttributes.IntegrationTypeEnum"
    },
    {
        "name": "catalogType",
        "baseName": "catalog_type",
        "type": "string"
    }
];
(function (CatalogCategoryCreateQueryResourceObjectAttributes) {
    let IntegrationTypeEnum;
    (function (IntegrationTypeEnum) {
        IntegrationTypeEnum[IntegrationTypeEnum["Custom"] = '$custom'] = "Custom";
    })(IntegrationTypeEnum = CatalogCategoryCreateQueryResourceObjectAttributes.IntegrationTypeEnum || (CatalogCategoryCreateQueryResourceObjectAttributes.IntegrationTypeEnum = {}));
})(CatalogCategoryCreateQueryResourceObjectAttributes = exports.CatalogCategoryCreateQueryResourceObjectAttributes || (exports.CatalogCategoryCreateQueryResourceObjectAttributes = {}));
