// klaviyo-api/dist/model/catalogVariantCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogVariantCreateQueryResourceObjectAttributes = void 0;
class CatalogVariantCreateQueryResourceObjectAttributes {
    constructor() {
        /**
        * The type of catalog. Currently only \"$default\" is supported.
        */
        this['catalogType'] = '$default';
        /**
        * The integration type. Currently only \"$custom\" is supported.
        */
        this['integrationType'] = CatalogVariantCreateQueryResourceObjectAttributes.IntegrationTypeEnum.Custom;
        /**
        * This field controls the visibility of this catalog item variant in product feeds/blocks. This field supports the following values: `1`: a product will not appear in dynamic product recommendation feeds and blocks if it is out of stock. `0` or `2`: a product can appear in dynamic product recommendation feeds and blocks regardless of inventory quantity.
        */
        this['inventoryPolicy'] = CatalogVariantCreateQueryResourceObjectAttributes.InventoryPolicyEnum.NUMBER_0;
        /**
        * Boolean value indicating whether the catalog item variant is published.
        */
        this['published'] = true;
    }
    static getAttributeTypeMap() {
        return CatalogVariantCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.CatalogVariantCreateQueryResourceObjectAttributes = CatalogVariantCreateQueryResourceObjectAttributes;
CatalogVariantCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "externalId",
        "baseName": "external_id",
        "type": "string"
    },
    {
        "name": "catalogType",
        "baseName": "catalog_type",
        "type": "string"
    },
    {
        "name": "integrationType",
        "baseName": "integration_type",
        "type": "CatalogVariantCreateQueryResourceObjectAttributes.IntegrationTypeEnum"
    },
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "sku",
        "baseName": "sku",
        "type": "string"
    },
    {
        "name": "inventoryPolicy",
        "baseName": "inventory_policy",
        "type": "CatalogVariantCreateQueryResourceObjectAttributes.InventoryPolicyEnum"
    },
    {
        "name": "inventoryQuantity",
        "baseName": "inventory_quantity",
        "type": "number"
    },
    {
        "name": "price",
        "baseName": "price",
        "type": "number"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "imageFullUrl",
        "baseName": "image_full_url",
        "type": "string"
    },
    {
        "name": "imageThumbnailUrl",
        "baseName": "image_thumbnail_url",
        "type": "string"
    },
    {
        "name": "images",
        "baseName": "images",
        "type": "Array<string>"
    },
    {
        "name": "customMetadata",
        "baseName": "custom_metadata",
        "type": "object"
    },
    {
        "name": "published",
        "baseName": "published",
        "type": "boolean"
    }
];
(function (CatalogVariantCreateQueryResourceObjectAttributes) {
    let IntegrationTypeEnum;
    (function (IntegrationTypeEnum) {
        IntegrationTypeEnum[IntegrationTypeEnum["Custom"] = '$custom'] = "Custom";
    })(IntegrationTypeEnum = CatalogVariantCreateQueryResourceObjectAttributes.IntegrationTypeEnum || (CatalogVariantCreateQueryResourceObjectAttributes.IntegrationTypeEnum = {}));
    let InventoryPolicyEnum;
    (function (InventoryPolicyEnum) {
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_0"] = 0] = "NUMBER_0";
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_1"] = 1] = "NUMBER_1";
        InventoryPolicyEnum[InventoryPolicyEnum["NUMBER_2"] = 2] = "NUMBER_2";
    })(InventoryPolicyEnum = CatalogVariantCreateQueryResourceObjectAttributes.InventoryPolicyEnum || (CatalogVariantCreateQueryResourceObjectAttributes.InventoryPolicyEnum = {}));
})(CatalogVariantCreateQueryResourceObjectAttributes = exports.CatalogVariantCreateQueryResourceObjectAttributes || (exports.CatalogVariantCreateQueryResourceObjectAttributes = {}));
