// klaviyo-api/dist/model/countdownDelayActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountdownDelayActionData = void 0;
class CountdownDelayActionData {
    constructor() {
        /**
        * Defined as FlowDateTrigger attributes in app.
        */
        this['unit'] = CountdownDelayActionData.UnitEnum.Days;
        this['value'] = 0;
        this['timezone'] = CountdownDelayActionData.TimezoneEnum.Profile;
    }
    static getAttributeTypeMap() {
        return CountdownDelayActionData.attributeTypeMap;
    }
}
exports.CountdownDelayActionData = CountdownDelayActionData;
CountdownDelayActionData.attributeTypeMap = [
    {
        "name": "unit",
        "baseName": "unit",
        "type": "CountdownDelayActionData.UnitEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    },
    {
        "name": "timezone",
        "baseName": "timezone",
        "type": "CountdownDelayActionData.TimezoneEnum"
    },
    {
        "name": "delayUntilTime",
        "baseName": "delay_until_time",
        "type": "string"
    },
    {
        "name": "delayUntilWeekdays",
        "baseName": "delay_until_weekdays",
        "type": "Array<CountdownDelayActionData.DelayUntilWeekdaysEnum>"
    }
];
(function (CountdownDelayActionData) {
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Days"] = 'days'] = "Days";
        UnitEnum[UnitEnum["Months"] = 'months'] = "Months";
        UnitEnum[UnitEnum["Weeks"] = 'weeks'] = "Weeks";
    })(UnitEnum = CountdownDelayActionData.UnitEnum || (CountdownDelayActionData.UnitEnum = {}));
    let TimezoneEnum;
    (function (TimezoneEnum) {
        TimezoneEnum[TimezoneEnum["AfricaAbidjan"] = 'Africa/Abidjan'] = "AfricaAbidjan";
        TimezoneEnum[TimezoneEnum["AfricaAccra"] = 'Africa/Accra'] = "AfricaAccra";
        TimezoneEnum[TimezoneEnum["AfricaAddisAbaba"] = 'Africa/Addis_Ababa'] = "AfricaAddisAbaba";
        TimezoneEnum[TimezoneEnum["AfricaAlgiers"] = 'Africa/Algiers'] = "AfricaAlgiers";
        TimezoneEnum[TimezoneEnum["AfricaAsmara"] = 'Africa/Asmara'] = "AfricaAsmara";
        TimezoneEnum[TimezoneEnum["AfricaBamako"] = 'Africa/Bamako'] = "AfricaBamako";
        TimezoneEnum[TimezoneEnum["AfricaBangui"] = 'Africa/Bangui'] = "AfricaBangui";
        TimezoneEnum[TimezoneEnum["AfricaBanjul"] = 'Africa/Banjul'] = "AfricaBanjul";
        TimezoneEnum[TimezoneEnum["AfricaBissau"] = 'Africa/Bissau'] = "AfricaBissau";
        TimezoneEnum[TimezoneEnum["AfricaBlantyre"] = 'Africa/Blantyre'] = "AfricaBlantyre";
        TimezoneEnum[TimezoneEnum["AfricaBrazzaville"] = 'Africa/Brazzaville'] = "AfricaBrazzaville";
        TimezoneEnum[TimezoneEnum["AfricaBujumbura"] = 'Africa/Bujumbura'] = "AfricaBujumbura";
        TimezoneEnum[TimezoneEnum["AfricaCairo"] = 'Africa/Cairo'] = "AfricaCairo";
        TimezoneEnum[TimezoneEnum["AfricaCasablanca"] = 'Africa/Casablanca'] = "AfricaCasablanca";
        TimezoneEnum[TimezoneEnum["AfricaCeuta"] = 'Africa/Ceuta'] = "AfricaCeuta";
        TimezoneEnum[TimezoneEnum["AfricaConakry"] = 'Africa/Conakry'] = "AfricaConakry";
        TimezoneEnum[TimezoneEnum["AfricaDakar"] = 'Africa/Dakar'] = "AfricaDakar";
        TimezoneEnum[TimezoneEnum["AfricaDarEsSalaam"] = 'Africa/Dar_es_Salaam'] = "AfricaDarEsSalaam";
        TimezoneEnum[TimezoneEnum["AfricaDjibouti"] = 'Africa/Djibouti'] = "AfricaDjibouti";
        TimezoneEnum[TimezoneEnum["AfricaDouala"] = 'Africa/Douala'] = "AfricaDouala";
        TimezoneEnum[TimezoneEnum["AfricaElAaiun"] = 'Africa/El_Aaiun'] = "AfricaElAaiun";
        TimezoneEnum[TimezoneEnum["AfricaFreetown"] = 'Africa/Freetown'] = "AfricaFreetown";
        TimezoneEnum[TimezoneEnum["AfricaGaborone"] = 'Africa/Gaborone'] = "AfricaGaborone";
        TimezoneEnum[TimezoneEnum["AfricaHarare"] = 'Africa/Harare'] = "AfricaHarare";
        TimezoneEnum[TimezoneEnum["AfricaJohannesburg"] = 'Africa/Johannesburg'] = "AfricaJohannesburg";
        TimezoneEnum[TimezoneEnum["AfricaJuba"] = 'Africa/Juba'] = "AfricaJuba";
        TimezoneEnum[TimezoneEnum["AfricaKampala"] = 'Africa/Kampala'] = "AfricaKampala";
        TimezoneEnum[TimezoneEnum["AfricaKhartoum"] = 'Africa/Khartoum'] = "AfricaKhartoum";
        TimezoneEnum[TimezoneEnum["AfricaKigali"] = 'Africa/Kigali'] = "AfricaKigali";
        TimezoneEnum[TimezoneEnum["AfricaKinshasa"] = 'Africa/Kinshasa'] = "AfricaKinshasa";
        TimezoneEnum[TimezoneEnum["AfricaLagos"] = 'Africa/Lagos'] = "AfricaLagos";
        TimezoneEnum[TimezoneEnum["AfricaLibreville"] = 'Africa/Libreville'] = "AfricaLibreville";
        TimezoneEnum[TimezoneEnum["AfricaLome"] = 'Africa/Lome'] = "AfricaLome";
        TimezoneEnum[TimezoneEnum["AfricaLuanda"] = 'Africa/Luanda'] = "AfricaLuanda";
        TimezoneEnum[TimezoneEnum["AfricaLubumbashi"] = 'Africa/Lubumbashi'] = "AfricaLubumbashi";
        TimezoneEnum[TimezoneEnum["AfricaLusaka"] = 'Africa/Lusaka'] = "AfricaLusaka";
        TimezoneEnum[TimezoneEnum["AfricaMalabo"] = 'Africa/Malabo'] = "AfricaMalabo";
        TimezoneEnum[TimezoneEnum["AfricaMaputo"] = 'Africa/Maputo'] = "AfricaMaputo";
        TimezoneEnum[TimezoneEnum["AfricaMaseru"] = 'Africa/Maseru'] = "AfricaMaseru";
        TimezoneEnum[TimezoneEnum["AfricaMbabane"] = 'Africa/Mbabane'] = "AfricaMbabane";
        TimezoneEnum[TimezoneEnum["AfricaMogadishu"] = 'Africa/Mogadishu'] = "AfricaMogadishu";
        TimezoneEnum[TimezoneEnum["AfricaMonrovia"] = 'Africa/Monrovia'] = "AfricaMonrovia";
        TimezoneEnum[TimezoneEnum["AfricaNairobi"] = 'Africa/Nairobi'] = "AfricaNairobi";
        TimezoneEnum[TimezoneEnum["AfricaNdjamena"] = 'Africa/Ndjamena'] = "AfricaNdjamena";
        TimezoneEnum[TimezoneEnum["AfricaNiamey"] = 'Africa/Niamey'] = "AfricaNiamey";
        TimezoneEnum[TimezoneEnum["AfricaNouakchott"] = 'Africa/Nouakchott'] = "AfricaNouakchott";
        TimezoneEnum[TimezoneEnum["AfricaOuagadougou"] = 'Africa/Ouagadougou'] = "AfricaOuagadougou";
        TimezoneEnum[TimezoneEnum["AfricaPortoNovo"] = 'Africa/Porto-Novo'] = "AfricaPortoNovo";
        TimezoneEnum[TimezoneEnum["AfricaSaoTome"] = 'Africa/Sao_Tome'] = "AfricaSaoTome";
        TimezoneEnum[TimezoneEnum["AfricaTripoli"] = 'Africa/Tripoli'] = "AfricaTripoli";
        TimezoneEnum[TimezoneEnum["AfricaTunis"] = 'Africa/Tunis'] = "AfricaTunis";
        TimezoneEnum[TimezoneEnum["AfricaWindhoek"] = 'Africa/Windhoek'] = "AfricaWindhoek";
        TimezoneEnum[TimezoneEnum["AmericaAdak"] = 'America/Adak'] = "AmericaAdak";
        TimezoneEnum[TimezoneEnum["AmericaAnchorage"] = 'America/Anchorage'] = "AmericaAnchorage";
        TimezoneEnum[TimezoneEnum["AmericaAnguilla"] = 'America/Anguilla'] = "AmericaAnguilla";
        TimezoneEnum[TimezoneEnum["AmericaAntigua"] = 'America/Antigua'] = "AmericaAntigua";
        TimezoneEnum[TimezoneEnum["AmericaAraguaina"] = 'America/Araguaina'] = "AmericaAraguaina";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaBuenosAires"] = 'America/Argentina/Buenos_Aires'] = "AmericaArgentinaBuenosAires";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaCatamarca"] = 'America/Argentina/Catamarca'] = "AmericaArgentinaCatamarca";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaCordoba"] = 'America/Argentina/Cordoba'] = "AmericaArgentinaCordoba";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaJujuy"] = 'America/Argentina/Jujuy'] = "AmericaArgentinaJujuy";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaLaRioja"] = 'America/Argentina/La_Rioja'] = "AmericaArgentinaLaRioja";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaMendoza"] = 'America/Argentina/Mendoza'] = "AmericaArgentinaMendoza";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaRioGallegos"] = 'America/Argentina/Rio_Gallegos'] = "AmericaArgentinaRioGallegos";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaSalta"] = 'America/Argentina/Salta'] = "AmericaArgentinaSalta";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaSanJuan"] = 'America/Argentina/San_Juan'] = "AmericaArgentinaSanJuan";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaSanLuis"] = 'America/Argentina/San_Luis'] = "AmericaArgentinaSanLuis";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaTucuman"] = 'America/Argentina/Tucuman'] = "AmericaArgentinaTucuman";
        TimezoneEnum[TimezoneEnum["AmericaArgentinaUshuaia"] = 'America/Argentina/Ushuaia'] = "AmericaArgentinaUshuaia";
        TimezoneEnum[TimezoneEnum["AmericaAruba"] = 'America/Aruba'] = "AmericaAruba";
        TimezoneEnum[TimezoneEnum["AmericaAsuncion"] = 'America/Asuncion'] = "AmericaAsuncion";
        TimezoneEnum[TimezoneEnum["AmericaAtikokan"] = 'America/Atikokan'] = "AmericaAtikokan";
        TimezoneEnum[TimezoneEnum["AmericaBahia"] = 'America/Bahia'] = "AmericaBahia";
        TimezoneEnum[TimezoneEnum["AmericaBahiaBanderas"] = 'America/Bahia_Banderas'] = "AmericaBahiaBanderas";
        TimezoneEnum[TimezoneEnum["AmericaBarbados"] = 'America/Barbados'] = "AmericaBarbados";
        TimezoneEnum[TimezoneEnum["AmericaBelem"] = 'America/Belem'] = "AmericaBelem";
        TimezoneEnum[TimezoneEnum["AmericaBelize"] = 'America/Belize'] = "AmericaBelize";
        TimezoneEnum[TimezoneEnum["AmericaBlancSablon"] = 'America/Blanc-Sablon'] = "AmericaBlancSablon";
        TimezoneEnum[TimezoneEnum["AmericaBoaVista"] = 'America/Boa_Vista'] = "AmericaBoaVista";
        TimezoneEnum[TimezoneEnum["AmericaBogota"] = 'America/Bogota'] = "AmericaBogota";
        TimezoneEnum[TimezoneEnum["AmericaBoise"] = 'America/Boise'] = "AmericaBoise";
        TimezoneEnum[TimezoneEnum["AmericaCambridgeBay"] = 'America/Cambridge_Bay'] = "AmericaCambridgeBay";
        TimezoneEnum[TimezoneEnum["AmericaCampoGrande"] = 'America/Campo_Grande'] = "AmericaCampoGrande";
        TimezoneEnum[TimezoneEnum["AmericaCancun"] = 'America/Cancun'] = "AmericaCancun";
        TimezoneEnum[TimezoneEnum["AmericaCaracas"] = 'America/Caracas'] = "AmericaCaracas";
        TimezoneEnum[TimezoneEnum["AmericaCayenne"] = 'America/Cayenne'] = "AmericaCayenne";
        TimezoneEnum[TimezoneEnum["AmericaCayman"] = 'America/Cayman'] = "AmericaCayman";
        TimezoneEnum[TimezoneEnum["AmericaChicago"] = 'America/Chicago'] = "AmericaChicago";
        TimezoneEnum[TimezoneEnum["AmericaChihuahua"] = 'America/Chihuahua'] = "AmericaChihuahua";
        TimezoneEnum[TimezoneEnum["AmericaCiudadJuarez"] = 'America/Ciudad_Juarez'] = "AmericaCiudadJuarez";
        TimezoneEnum[TimezoneEnum["AmericaCostaRica"] = 'America/Costa_Rica'] = "AmericaCostaRica";
        TimezoneEnum[TimezoneEnum["AmericaCreston"] = 'America/Creston'] = "AmericaCreston";
        TimezoneEnum[TimezoneEnum["AmericaCuiaba"] = 'America/Cuiaba'] = "AmericaCuiaba";
        TimezoneEnum[TimezoneEnum["AmericaCuracao"] = 'America/Curacao'] = "AmericaCuracao";
        TimezoneEnum[TimezoneEnum["AmericaDanmarkshavn"] = 'America/Danmarkshavn'] = "AmericaDanmarkshavn";
        TimezoneEnum[TimezoneEnum["AmericaDawson"] = 'America/Dawson'] = "AmericaDawson";
        TimezoneEnum[TimezoneEnum["AmericaDawsonCreek"] = 'America/Dawson_Creek'] = "AmericaDawsonCreek";
        TimezoneEnum[TimezoneEnum["AmericaDenver"] = 'America/Denver'] = "AmericaDenver";
        TimezoneEnum[TimezoneEnum["AmericaDetroit"] = 'America/Detroit'] = "AmericaDetroit";
        TimezoneEnum[TimezoneEnum["AmericaDominica"] = 'America/Dominica'] = "AmericaDominica";
        TimezoneEnum[TimezoneEnum["AmericaEdmonton"] = 'America/Edmonton'] = "AmericaEdmonton";
        TimezoneEnum[TimezoneEnum["AmericaEirunepe"] = 'America/Eirunepe'] = "AmericaEirunepe";
        TimezoneEnum[TimezoneEnum["AmericaElSalvador"] = 'America/El_Salvador'] = "AmericaElSalvador";
        TimezoneEnum[TimezoneEnum["AmericaFortNelson"] = 'America/Fort_Nelson'] = "AmericaFortNelson";
        TimezoneEnum[TimezoneEnum["AmericaFortaleza"] = 'America/Fortaleza'] = "AmericaFortaleza";
        TimezoneEnum[TimezoneEnum["AmericaGlaceBay"] = 'America/Glace_Bay'] = "AmericaGlaceBay";
        TimezoneEnum[TimezoneEnum["AmericaGooseBay"] = 'America/Goose_Bay'] = "AmericaGooseBay";
        TimezoneEnum[TimezoneEnum["AmericaGrandTurk"] = 'America/Grand_Turk'] = "AmericaGrandTurk";
        TimezoneEnum[TimezoneEnum["AmericaGrenada"] = 'America/Grenada'] = "AmericaGrenada";
        TimezoneEnum[TimezoneEnum["AmericaGuadeloupe"] = 'America/Guadeloupe'] = "AmericaGuadeloupe";
        TimezoneEnum[TimezoneEnum["AmericaGuatemala"] = 'America/Guatemala'] = "AmericaGuatemala";
        TimezoneEnum[TimezoneEnum["AmericaGuayaquil"] = 'America/Guayaquil'] = "AmericaGuayaquil";
        TimezoneEnum[TimezoneEnum["AmericaGuyana"] = 'America/Guyana'] = "AmericaGuyana";
        TimezoneEnum[TimezoneEnum["AmericaHalifax"] = 'America/Halifax'] = "AmericaHalifax";
        TimezoneEnum[TimezoneEnum["AmericaHavana"] = 'America/Havana'] = "AmericaHavana";
        TimezoneEnum[TimezoneEnum["AmericaHermosillo"] = 'America/Hermosillo'] = "AmericaHermosillo";
        TimezoneEnum[TimezoneEnum["AmericaIndianaIndianapolis"] = 'America/Indiana/Indianapolis'] = "AmericaIndianaIndianapolis";
        TimezoneEnum[TimezoneEnum["AmericaIndianaKnox"] = 'America/Indiana/Knox'] = "AmericaIndianaKnox";
        TimezoneEnum[TimezoneEnum["AmericaIndianaMarengo"] = 'America/Indiana/Marengo'] = "AmericaIndianaMarengo";
        TimezoneEnum[TimezoneEnum["AmericaIndianaPetersburg"] = 'America/Indiana/Petersburg'] = "AmericaIndianaPetersburg";
        TimezoneEnum[TimezoneEnum["AmericaIndianaTellCity"] = 'America/Indiana/Tell_City'] = "AmericaIndianaTellCity";
        TimezoneEnum[TimezoneEnum["AmericaIndianaVevay"] = 'America/Indiana/Vevay'] = "AmericaIndianaVevay";
        TimezoneEnum[TimezoneEnum["AmericaIndianaVincennes"] = 'America/Indiana/Vincennes'] = "AmericaIndianaVincennes";
        TimezoneEnum[TimezoneEnum["AmericaIndianaWinamac"] = 'America/Indiana/Winamac'] = "AmericaIndianaWinamac";
        TimezoneEnum[TimezoneEnum["AmericaInuvik"] = 'America/Inuvik'] = "AmericaInuvik";
        TimezoneEnum[TimezoneEnum["AmericaIqaluit"] = 'America/Iqaluit'] = "AmericaIqaluit";
        TimezoneEnum[TimezoneEnum["AmericaJamaica"] = 'America/Jamaica'] = "AmericaJamaica";
        TimezoneEnum[TimezoneEnum["AmericaJuneau"] = 'America/Juneau'] = "AmericaJuneau";
        TimezoneEnum[TimezoneEnum["AmericaKentuckyLouisville"] = 'America/Kentucky/Louisville'] = "AmericaKentuckyLouisville";
        TimezoneEnum[TimezoneEnum["AmericaKentuckyMonticello"] = 'America/Kentucky/Monticello'] = "AmericaKentuckyMonticello";
        TimezoneEnum[TimezoneEnum["AmericaKralendijk"] = 'America/Kralendijk'] = "AmericaKralendijk";
        TimezoneEnum[TimezoneEnum["AmericaLaPaz"] = 'America/La_Paz'] = "AmericaLaPaz";
        TimezoneEnum[TimezoneEnum["AmericaLima"] = 'America/Lima'] = "AmericaLima";
        TimezoneEnum[TimezoneEnum["AmericaLosAngeles"] = 'America/Los_Angeles'] = "AmericaLosAngeles";
        TimezoneEnum[TimezoneEnum["AmericaLowerPrinces"] = 'America/Lower_Princes'] = "AmericaLowerPrinces";
        TimezoneEnum[TimezoneEnum["AmericaMaceio"] = 'America/Maceio'] = "AmericaMaceio";
        TimezoneEnum[TimezoneEnum["AmericaManagua"] = 'America/Managua'] = "AmericaManagua";
        TimezoneEnum[TimezoneEnum["AmericaManaus"] = 'America/Manaus'] = "AmericaManaus";
        TimezoneEnum[TimezoneEnum["AmericaMarigot"] = 'America/Marigot'] = "AmericaMarigot";
        TimezoneEnum[TimezoneEnum["AmericaMartinique"] = 'America/Martinique'] = "AmericaMartinique";
        TimezoneEnum[TimezoneEnum["AmericaMatamoros"] = 'America/Matamoros'] = "AmericaMatamoros";
        TimezoneEnum[TimezoneEnum["AmericaMazatlan"] = 'America/Mazatlan'] = "AmericaMazatlan";
        TimezoneEnum[TimezoneEnum["AmericaMenominee"] = 'America/Menominee'] = "AmericaMenominee";
        TimezoneEnum[TimezoneEnum["AmericaMerida"] = 'America/Merida'] = "AmericaMerida";
        TimezoneEnum[TimezoneEnum["AmericaMetlakatla"] = 'America/Metlakatla'] = "AmericaMetlakatla";
        TimezoneEnum[TimezoneEnum["AmericaMexicoCity"] = 'America/Mexico_City'] = "AmericaMexicoCity";
        TimezoneEnum[TimezoneEnum["AmericaMiquelon"] = 'America/Miquelon'] = "AmericaMiquelon";
        TimezoneEnum[TimezoneEnum["AmericaMoncton"] = 'America/Moncton'] = "AmericaMoncton";
        TimezoneEnum[TimezoneEnum["AmericaMonterrey"] = 'America/Monterrey'] = "AmericaMonterrey";
        TimezoneEnum[TimezoneEnum["AmericaMontevideo"] = 'America/Montevideo'] = "AmericaMontevideo";
        TimezoneEnum[TimezoneEnum["AmericaMontserrat"] = 'America/Montserrat'] = "AmericaMontserrat";
        TimezoneEnum[TimezoneEnum["AmericaNassau"] = 'America/Nassau'] = "AmericaNassau";
        TimezoneEnum[TimezoneEnum["AmericaNewYork"] = 'America/New_York'] = "AmericaNewYork";
        TimezoneEnum[TimezoneEnum["AmericaNome"] = 'America/Nome'] = "AmericaNome";
        TimezoneEnum[TimezoneEnum["AmericaNoronha"] = 'America/Noronha'] = "AmericaNoronha";
        TimezoneEnum[TimezoneEnum["AmericaNorthDakotaBeulah"] = 'America/North_Dakota/Beulah'] = "AmericaNorthDakotaBeulah";
        TimezoneEnum[TimezoneEnum["AmericaNorthDakotaCenter"] = 'America/North_Dakota/Center'] = "AmericaNorthDakotaCenter";
        TimezoneEnum[TimezoneEnum["AmericaNorthDakotaNewSalem"] = 'America/North_Dakota/New_Salem'] = "AmericaNorthDakotaNewSalem";
        TimezoneEnum[TimezoneEnum["AmericaNuuk"] = 'America/Nuuk'] = "AmericaNuuk";
        TimezoneEnum[TimezoneEnum["AmericaOjinaga"] = 'America/Ojinaga'] = "AmericaOjinaga";
        TimezoneEnum[TimezoneEnum["AmericaPanama"] = 'America/Panama'] = "AmericaPanama";
        TimezoneEnum[TimezoneEnum["AmericaParamaribo"] = 'America/Paramaribo'] = "AmericaParamaribo";
        TimezoneEnum[TimezoneEnum["AmericaPhoenix"] = 'America/Phoenix'] = "AmericaPhoenix";
        TimezoneEnum[TimezoneEnum["AmericaPortAuPrince"] = 'America/Port-au-Prince'] = "AmericaPortAuPrince";
        TimezoneEnum[TimezoneEnum["AmericaPortOfSpain"] = 'America/Port_of_Spain'] = "AmericaPortOfSpain";
        TimezoneEnum[TimezoneEnum["AmericaPortoVelho"] = 'America/Porto_Velho'] = "AmericaPortoVelho";
        TimezoneEnum[TimezoneEnum["AmericaPuertoRico"] = 'America/Puerto_Rico'] = "AmericaPuertoRico";
        TimezoneEnum[TimezoneEnum["AmericaPuntaArenas"] = 'America/Punta_Arenas'] = "AmericaPuntaArenas";
        TimezoneEnum[TimezoneEnum["AmericaRankinInlet"] = 'America/Rankin_Inlet'] = "AmericaRankinInlet";
        TimezoneEnum[TimezoneEnum["AmericaRecife"] = 'America/Recife'] = "AmericaRecife";
        TimezoneEnum[TimezoneEnum["AmericaRegina"] = 'America/Regina'] = "AmericaRegina";
        TimezoneEnum[TimezoneEnum["AmericaResolute"] = 'America/Resolute'] = "AmericaResolute";
        TimezoneEnum[TimezoneEnum["AmericaRioBranco"] = 'America/Rio_Branco'] = "AmericaRioBranco";
        TimezoneEnum[TimezoneEnum["AmericaSantarem"] = 'America/Santarem'] = "AmericaSantarem";
        TimezoneEnum[TimezoneEnum["AmericaSantiago"] = 'America/Santiago'] = "AmericaSantiago";
        TimezoneEnum[TimezoneEnum["AmericaSantoDomingo"] = 'America/Santo_Domingo'] = "AmericaSantoDomingo";
        TimezoneEnum[TimezoneEnum["AmericaSaoPaulo"] = 'America/Sao_Paulo'] = "AmericaSaoPaulo";
        TimezoneEnum[TimezoneEnum["AmericaScoresbysund"] = 'America/Scoresbysund'] = "AmericaScoresbysund";
        TimezoneEnum[TimezoneEnum["AmericaSitka"] = 'America/Sitka'] = "AmericaSitka";
        TimezoneEnum[TimezoneEnum["AmericaStBarthelemy"] = 'America/St_Barthelemy'] = "AmericaStBarthelemy";
        TimezoneEnum[TimezoneEnum["AmericaStJohns"] = 'America/St_Johns'] = "AmericaStJohns";
        TimezoneEnum[TimezoneEnum["AmericaStKitts"] = 'America/St_Kitts'] = "AmericaStKitts";
        TimezoneEnum[TimezoneEnum["AmericaStLucia"] = 'America/St_Lucia'] = "AmericaStLucia";
        TimezoneEnum[TimezoneEnum["AmericaStThomas"] = 'America/St_Thomas'] = "AmericaStThomas";
        TimezoneEnum[TimezoneEnum["AmericaStVincent"] = 'America/St_Vincent'] = "AmericaStVincent";
        TimezoneEnum[TimezoneEnum["AmericaSwiftCurrent"] = 'America/Swift_Current'] = "AmericaSwiftCurrent";
        TimezoneEnum[TimezoneEnum["AmericaTegucigalpa"] = 'America/Tegucigalpa'] = "AmericaTegucigalpa";
        TimezoneEnum[TimezoneEnum["AmericaThule"] = 'America/Thule'] = "AmericaThule";
        TimezoneEnum[TimezoneEnum["AmericaTijuana"] = 'America/Tijuana'] = "AmericaTijuana";
        TimezoneEnum[TimezoneEnum["AmericaToronto"] = 'America/Toronto'] = "AmericaToronto";
        TimezoneEnum[TimezoneEnum["AmericaTortola"] = 'America/Tortola'] = "AmericaTortola";
        TimezoneEnum[TimezoneEnum["AmericaVancouver"] = 'America/Vancouver'] = "AmericaVancouver";
        TimezoneEnum[TimezoneEnum["AmericaWhitehorse"] = 'America/Whitehorse'] = "AmericaWhitehorse";
        TimezoneEnum[TimezoneEnum["AmericaWinnipeg"] = 'America/Winnipeg'] = "AmericaWinnipeg";
        TimezoneEnum[TimezoneEnum["AmericaYakutat"] = 'America/Yakutat'] = "AmericaYakutat";
        TimezoneEnum[TimezoneEnum["AntarcticaCasey"] = 'Antarctica/Casey'] = "AntarcticaCasey";
        TimezoneEnum[TimezoneEnum["AntarcticaDavis"] = 'Antarctica/Davis'] = "AntarcticaDavis";
        TimezoneEnum[TimezoneEnum["AntarcticaDumontDUrville"] = 'Antarctica/DumontDUrville'] = "AntarcticaDumontDUrville";
        TimezoneEnum[TimezoneEnum["AntarcticaMacquarie"] = 'Antarctica/Macquarie'] = "AntarcticaMacquarie";
        TimezoneEnum[TimezoneEnum["AntarcticaMawson"] = 'Antarctica/Mawson'] = "AntarcticaMawson";
        TimezoneEnum[TimezoneEnum["AntarcticaMcMurdo"] = 'Antarctica/McMurdo'] = "AntarcticaMcMurdo";
        TimezoneEnum[TimezoneEnum["AntarcticaPalmer"] = 'Antarctica/Palmer'] = "AntarcticaPalmer";
        TimezoneEnum[TimezoneEnum["AntarcticaRothera"] = 'Antarctica/Rothera'] = "AntarcticaRothera";
        TimezoneEnum[TimezoneEnum["AntarcticaSyowa"] = 'Antarctica/Syowa'] = "AntarcticaSyowa";
        TimezoneEnum[TimezoneEnum["AntarcticaTroll"] = 'Antarctica/Troll'] = "AntarcticaTroll";
        TimezoneEnum[TimezoneEnum["AntarcticaVostok"] = 'Antarctica/Vostok'] = "AntarcticaVostok";
        TimezoneEnum[TimezoneEnum["ArcticLongyearbyen"] = 'Arctic/Longyearbyen'] = "ArcticLongyearbyen";
        TimezoneEnum[TimezoneEnum["AsiaAden"] = 'Asia/Aden'] = "AsiaAden";
        TimezoneEnum[TimezoneEnum["AsiaAlmaty"] = 'Asia/Almaty'] = "AsiaAlmaty";
        TimezoneEnum[TimezoneEnum["AsiaAmman"] = 'Asia/Amman'] = "AsiaAmman";
        TimezoneEnum[TimezoneEnum["AsiaAnadyr"] = 'Asia/Anadyr'] = "AsiaAnadyr";
        TimezoneEnum[TimezoneEnum["AsiaAqtau"] = 'Asia/Aqtau'] = "AsiaAqtau";
        TimezoneEnum[TimezoneEnum["AsiaAqtobe"] = 'Asia/Aqtobe'] = "AsiaAqtobe";
        TimezoneEnum[TimezoneEnum["AsiaAshgabat"] = 'Asia/Ashgabat'] = "AsiaAshgabat";
        TimezoneEnum[TimezoneEnum["AsiaAtyrau"] = 'Asia/Atyrau'] = "AsiaAtyrau";
        TimezoneEnum[TimezoneEnum["AsiaBaghdad"] = 'Asia/Baghdad'] = "AsiaBaghdad";
        TimezoneEnum[TimezoneEnum["AsiaBahrain"] = 'Asia/Bahrain'] = "AsiaBahrain";
        TimezoneEnum[TimezoneEnum["AsiaBaku"] = 'Asia/Baku'] = "AsiaBaku";
        TimezoneEnum[TimezoneEnum["AsiaBangkok"] = 'Asia/Bangkok'] = "AsiaBangkok";
        TimezoneEnum[TimezoneEnum["AsiaBarnaul"] = 'Asia/Barnaul'] = "AsiaBarnaul";
        TimezoneEnum[TimezoneEnum["AsiaBeirut"] = 'Asia/Beirut'] = "AsiaBeirut";
        TimezoneEnum[TimezoneEnum["AsiaBishkek"] = 'Asia/Bishkek'] = "AsiaBishkek";
        TimezoneEnum[TimezoneEnum["AsiaBrunei"] = 'Asia/Brunei'] = "AsiaBrunei";
        TimezoneEnum[TimezoneEnum["AsiaChita"] = 'Asia/Chita'] = "AsiaChita";
        TimezoneEnum[TimezoneEnum["AsiaChoibalsan"] = 'Asia/Choibalsan'] = "AsiaChoibalsan";
        TimezoneEnum[TimezoneEnum["AsiaColombo"] = 'Asia/Colombo'] = "AsiaColombo";
        TimezoneEnum[TimezoneEnum["AsiaDamascus"] = 'Asia/Damascus'] = "AsiaDamascus";
        TimezoneEnum[TimezoneEnum["AsiaDhaka"] = 'Asia/Dhaka'] = "AsiaDhaka";
        TimezoneEnum[TimezoneEnum["AsiaDili"] = 'Asia/Dili'] = "AsiaDili";
        TimezoneEnum[TimezoneEnum["AsiaDubai"] = 'Asia/Dubai'] = "AsiaDubai";
        TimezoneEnum[TimezoneEnum["AsiaDushanbe"] = 'Asia/Dushanbe'] = "AsiaDushanbe";
        TimezoneEnum[TimezoneEnum["AsiaFamagusta"] = 'Asia/Famagusta'] = "AsiaFamagusta";
        TimezoneEnum[TimezoneEnum["AsiaGaza"] = 'Asia/Gaza'] = "AsiaGaza";
        TimezoneEnum[TimezoneEnum["AsiaHebron"] = 'Asia/Hebron'] = "AsiaHebron";
        TimezoneEnum[TimezoneEnum["AsiaHoChiMinh"] = 'Asia/Ho_Chi_Minh'] = "AsiaHoChiMinh";
        TimezoneEnum[TimezoneEnum["AsiaHongKong"] = 'Asia/Hong_Kong'] = "AsiaHongKong";
        TimezoneEnum[TimezoneEnum["AsiaHovd"] = 'Asia/Hovd'] = "AsiaHovd";
        TimezoneEnum[TimezoneEnum["AsiaIrkutsk"] = 'Asia/Irkutsk'] = "AsiaIrkutsk";
        TimezoneEnum[TimezoneEnum["AsiaJakarta"] = 'Asia/Jakarta'] = "AsiaJakarta";
        TimezoneEnum[TimezoneEnum["AsiaJayapura"] = 'Asia/Jayapura'] = "AsiaJayapura";
        TimezoneEnum[TimezoneEnum["AsiaJerusalem"] = 'Asia/Jerusalem'] = "AsiaJerusalem";
        TimezoneEnum[TimezoneEnum["AsiaKabul"] = 'Asia/Kabul'] = "AsiaKabul";
        TimezoneEnum[TimezoneEnum["AsiaKamchatka"] = 'Asia/Kamchatka'] = "AsiaKamchatka";
        TimezoneEnum[TimezoneEnum["AsiaKarachi"] = 'Asia/Karachi'] = "AsiaKarachi";
        TimezoneEnum[TimezoneEnum["AsiaKathmandu"] = 'Asia/Kathmandu'] = "AsiaKathmandu";
        TimezoneEnum[TimezoneEnum["AsiaKhandyga"] = 'Asia/Khandyga'] = "AsiaKhandyga";
        TimezoneEnum[TimezoneEnum["AsiaKolkata"] = 'Asia/Kolkata'] = "AsiaKolkata";
        TimezoneEnum[TimezoneEnum["AsiaKrasnoyarsk"] = 'Asia/Krasnoyarsk'] = "AsiaKrasnoyarsk";
        TimezoneEnum[TimezoneEnum["AsiaKualaLumpur"] = 'Asia/Kuala_Lumpur'] = "AsiaKualaLumpur";
        TimezoneEnum[TimezoneEnum["AsiaKuching"] = 'Asia/Kuching'] = "AsiaKuching";
        TimezoneEnum[TimezoneEnum["AsiaKuwait"] = 'Asia/Kuwait'] = "AsiaKuwait";
        TimezoneEnum[TimezoneEnum["AsiaMacau"] = 'Asia/Macau'] = "AsiaMacau";
        TimezoneEnum[TimezoneEnum["AsiaMagadan"] = 'Asia/Magadan'] = "AsiaMagadan";
        TimezoneEnum[TimezoneEnum["AsiaMakassar"] = 'Asia/Makassar'] = "AsiaMakassar";
        TimezoneEnum[TimezoneEnum["AsiaManila"] = 'Asia/Manila'] = "AsiaManila";
        TimezoneEnum[TimezoneEnum["AsiaMuscat"] = 'Asia/Muscat'] = "AsiaMuscat";
        TimezoneEnum[TimezoneEnum["AsiaNicosia"] = 'Asia/Nicosia'] = "AsiaNicosia";
        TimezoneEnum[TimezoneEnum["AsiaNovokuznetsk"] = 'Asia/Novokuznetsk'] = "AsiaNovokuznetsk";
        TimezoneEnum[TimezoneEnum["AsiaNovosibirsk"] = 'Asia/Novosibirsk'] = "AsiaNovosibirsk";
        TimezoneEnum[TimezoneEnum["AsiaOmsk"] = 'Asia/Omsk'] = "AsiaOmsk";
        TimezoneEnum[TimezoneEnum["AsiaOral"] = 'Asia/Oral'] = "AsiaOral";
        TimezoneEnum[TimezoneEnum["AsiaPhnomPenh"] = 'Asia/Phnom_Penh'] = "AsiaPhnomPenh";
        TimezoneEnum[TimezoneEnum["AsiaPontianak"] = 'Asia/Pontianak'] = "AsiaPontianak";
        TimezoneEnum[TimezoneEnum["AsiaPyongyang"] = 'Asia/Pyongyang'] = "AsiaPyongyang";
        TimezoneEnum[TimezoneEnum["AsiaQatar"] = 'Asia/Qatar'] = "AsiaQatar";
        TimezoneEnum[TimezoneEnum["AsiaQostanay"] = 'Asia/Qostanay'] = "AsiaQostanay";
        TimezoneEnum[TimezoneEnum["AsiaQyzylorda"] = 'Asia/Qyzylorda'] = "AsiaQyzylorda";
        TimezoneEnum[TimezoneEnum["AsiaRiyadh"] = 'Asia/Riyadh'] = "AsiaRiyadh";
        TimezoneEnum[TimezoneEnum["AsiaSakhalin"] = 'Asia/Sakhalin'] = "AsiaSakhalin";
        TimezoneEnum[TimezoneEnum["AsiaSamarkand"] = 'Asia/Samarkand'] = "AsiaSamarkand";
        TimezoneEnum[TimezoneEnum["AsiaSeoul"] = 'Asia/Seoul'] = "AsiaSeoul";
        TimezoneEnum[TimezoneEnum["AsiaShanghai"] = 'Asia/Shanghai'] = "AsiaShanghai";
        TimezoneEnum[TimezoneEnum["AsiaSingapore"] = 'Asia/Singapore'] = "AsiaSingapore";
        TimezoneEnum[TimezoneEnum["AsiaSrednekolymsk"] = 'Asia/Srednekolymsk'] = "AsiaSrednekolymsk";
        TimezoneEnum[TimezoneEnum["AsiaTaipei"] = 'Asia/Taipei'] = "AsiaTaipei";
        TimezoneEnum[TimezoneEnum["AsiaTashkent"] = 'Asia/Tashkent'] = "AsiaTashkent";
        TimezoneEnum[TimezoneEnum["AsiaTbilisi"] = 'Asia/Tbilisi'] = "AsiaTbilisi";
        TimezoneEnum[TimezoneEnum["AsiaTehran"] = 'Asia/Tehran'] = "AsiaTehran";
        TimezoneEnum[TimezoneEnum["AsiaThimphu"] = 'Asia/Thimphu'] = "AsiaThimphu";
        TimezoneEnum[TimezoneEnum["AsiaTokyo"] = 'Asia/Tokyo'] = "AsiaTokyo";
        TimezoneEnum[TimezoneEnum["AsiaTomsk"] = 'Asia/Tomsk'] = "AsiaTomsk";
        TimezoneEnum[TimezoneEnum["AsiaUlaanbaatar"] = 'Asia/Ulaanbaatar'] = "AsiaUlaanbaatar";
        TimezoneEnum[TimezoneEnum["AsiaUrumqi"] = 'Asia/Urumqi'] = "AsiaUrumqi";
        TimezoneEnum[TimezoneEnum["AsiaUstNera"] = 'Asia/Ust-Nera'] = "AsiaUstNera";
        TimezoneEnum[TimezoneEnum["AsiaVientiane"] = 'Asia/Vientiane'] = "AsiaVientiane";
        TimezoneEnum[TimezoneEnum["AsiaVladivostok"] = 'Asia/Vladivostok'] = "AsiaVladivostok";
        TimezoneEnum[TimezoneEnum["AsiaYakutsk"] = 'Asia/Yakutsk'] = "AsiaYakutsk";
        TimezoneEnum[TimezoneEnum["AsiaYangon"] = 'Asia/Yangon'] = "AsiaYangon";
        TimezoneEnum[TimezoneEnum["AsiaYekaterinburg"] = 'Asia/Yekaterinburg'] = "AsiaYekaterinburg";
        TimezoneEnum[TimezoneEnum["AsiaYerevan"] = 'Asia/Yerevan'] = "AsiaYerevan";
        TimezoneEnum[TimezoneEnum["AtlanticAzores"] = 'Atlantic/Azores'] = "AtlanticAzores";
        TimezoneEnum[TimezoneEnum["AtlanticBermuda"] = 'Atlantic/Bermuda'] = "AtlanticBermuda";
        TimezoneEnum[TimezoneEnum["AtlanticCanary"] = 'Atlantic/Canary'] = "AtlanticCanary";
        TimezoneEnum[TimezoneEnum["AtlanticCapeVerde"] = 'Atlantic/Cape_Verde'] = "AtlanticCapeVerde";
        TimezoneEnum[TimezoneEnum["AtlanticFaroe"] = 'Atlantic/Faroe'] = "AtlanticFaroe";
        TimezoneEnum[TimezoneEnum["AtlanticMadeira"] = 'Atlantic/Madeira'] = "AtlanticMadeira";
        TimezoneEnum[TimezoneEnum["AtlanticReykjavik"] = 'Atlantic/Reykjavik'] = "AtlanticReykjavik";
        TimezoneEnum[TimezoneEnum["AtlanticSouthGeorgia"] = 'Atlantic/South_Georgia'] = "AtlanticSouthGeorgia";
        TimezoneEnum[TimezoneEnum["AtlanticStHelena"] = 'Atlantic/St_Helena'] = "AtlanticStHelena";
        TimezoneEnum[TimezoneEnum["AtlanticStanley"] = 'Atlantic/Stanley'] = "AtlanticStanley";
        TimezoneEnum[TimezoneEnum["AustraliaAdelaide"] = 'Australia/Adelaide'] = "AustraliaAdelaide";
        TimezoneEnum[TimezoneEnum["AustraliaBrisbane"] = 'Australia/Brisbane'] = "AustraliaBrisbane";
        TimezoneEnum[TimezoneEnum["AustraliaBrokenHill"] = 'Australia/Broken_Hill'] = "AustraliaBrokenHill";
        TimezoneEnum[TimezoneEnum["AustraliaDarwin"] = 'Australia/Darwin'] = "AustraliaDarwin";
        TimezoneEnum[TimezoneEnum["AustraliaEucla"] = 'Australia/Eucla'] = "AustraliaEucla";
        TimezoneEnum[TimezoneEnum["AustraliaHobart"] = 'Australia/Hobart'] = "AustraliaHobart";
        TimezoneEnum[TimezoneEnum["AustraliaLindeman"] = 'Australia/Lindeman'] = "AustraliaLindeman";
        TimezoneEnum[TimezoneEnum["AustraliaLordHowe"] = 'Australia/Lord_Howe'] = "AustraliaLordHowe";
        TimezoneEnum[TimezoneEnum["AustraliaMelbourne"] = 'Australia/Melbourne'] = "AustraliaMelbourne";
        TimezoneEnum[TimezoneEnum["AustraliaPerth"] = 'Australia/Perth'] = "AustraliaPerth";
        TimezoneEnum[TimezoneEnum["AustraliaSydney"] = 'Australia/Sydney'] = "AustraliaSydney";
        TimezoneEnum[TimezoneEnum["CanadaAtlantic"] = 'Canada/Atlantic'] = "CanadaAtlantic";
        TimezoneEnum[TimezoneEnum["CanadaCentral"] = 'Canada/Central'] = "CanadaCentral";
        TimezoneEnum[TimezoneEnum["CanadaEastern"] = 'Canada/Eastern'] = "CanadaEastern";
        TimezoneEnum[TimezoneEnum["CanadaMountain"] = 'Canada/Mountain'] = "CanadaMountain";
        TimezoneEnum[TimezoneEnum["CanadaNewfoundland"] = 'Canada/Newfoundland'] = "CanadaNewfoundland";
        TimezoneEnum[TimezoneEnum["CanadaPacific"] = 'Canada/Pacific'] = "CanadaPacific";
        TimezoneEnum[TimezoneEnum["EuropeAmsterdam"] = 'Europe/Amsterdam'] = "EuropeAmsterdam";
        TimezoneEnum[TimezoneEnum["EuropeAndorra"] = 'Europe/Andorra'] = "EuropeAndorra";
        TimezoneEnum[TimezoneEnum["EuropeAstrakhan"] = 'Europe/Astrakhan'] = "EuropeAstrakhan";
        TimezoneEnum[TimezoneEnum["EuropeAthens"] = 'Europe/Athens'] = "EuropeAthens";
        TimezoneEnum[TimezoneEnum["EuropeBelgrade"] = 'Europe/Belgrade'] = "EuropeBelgrade";
        TimezoneEnum[TimezoneEnum["EuropeBerlin"] = 'Europe/Berlin'] = "EuropeBerlin";
        TimezoneEnum[TimezoneEnum["EuropeBratislava"] = 'Europe/Bratislava'] = "EuropeBratislava";
        TimezoneEnum[TimezoneEnum["EuropeBrussels"] = 'Europe/Brussels'] = "EuropeBrussels";
        TimezoneEnum[TimezoneEnum["EuropeBucharest"] = 'Europe/Bucharest'] = "EuropeBucharest";
        TimezoneEnum[TimezoneEnum["EuropeBudapest"] = 'Europe/Budapest'] = "EuropeBudapest";
        TimezoneEnum[TimezoneEnum["EuropeBusingen"] = 'Europe/Busingen'] = "EuropeBusingen";
        TimezoneEnum[TimezoneEnum["EuropeChisinau"] = 'Europe/Chisinau'] = "EuropeChisinau";
        TimezoneEnum[TimezoneEnum["EuropeCopenhagen"] = 'Europe/Copenhagen'] = "EuropeCopenhagen";
        TimezoneEnum[TimezoneEnum["EuropeDublin"] = 'Europe/Dublin'] = "EuropeDublin";
        TimezoneEnum[TimezoneEnum["EuropeGibraltar"] = 'Europe/Gibraltar'] = "EuropeGibraltar";
        TimezoneEnum[TimezoneEnum["EuropeGuernsey"] = 'Europe/Guernsey'] = "EuropeGuernsey";
        TimezoneEnum[TimezoneEnum["EuropeHelsinki"] = 'Europe/Helsinki'] = "EuropeHelsinki";
        TimezoneEnum[TimezoneEnum["EuropeIsleOfMan"] = 'Europe/Isle_of_Man'] = "EuropeIsleOfMan";
        TimezoneEnum[TimezoneEnum["EuropeIstanbul"] = 'Europe/Istanbul'] = "EuropeIstanbul";
        TimezoneEnum[TimezoneEnum["EuropeJersey"] = 'Europe/Jersey'] = "EuropeJersey";
        TimezoneEnum[TimezoneEnum["EuropeKaliningrad"] = 'Europe/Kaliningrad'] = "EuropeKaliningrad";
        TimezoneEnum[TimezoneEnum["EuropeKirov"] = 'Europe/Kirov'] = "EuropeKirov";
        TimezoneEnum[TimezoneEnum["EuropeKyiv"] = 'Europe/Kyiv'] = "EuropeKyiv";
        TimezoneEnum[TimezoneEnum["EuropeLisbon"] = 'Europe/Lisbon'] = "EuropeLisbon";
        TimezoneEnum[TimezoneEnum["EuropeLjubljana"] = 'Europe/Ljubljana'] = "EuropeLjubljana";
        TimezoneEnum[TimezoneEnum["EuropeLondon"] = 'Europe/London'] = "EuropeLondon";
        TimezoneEnum[TimezoneEnum["EuropeLuxembourg"] = 'Europe/Luxembourg'] = "EuropeLuxembourg";
        TimezoneEnum[TimezoneEnum["EuropeMadrid"] = 'Europe/Madrid'] = "EuropeMadrid";
        TimezoneEnum[TimezoneEnum["EuropeMalta"] = 'Europe/Malta'] = "EuropeMalta";
        TimezoneEnum[TimezoneEnum["EuropeMariehamn"] = 'Europe/Mariehamn'] = "EuropeMariehamn";
        TimezoneEnum[TimezoneEnum["EuropeMinsk"] = 'Europe/Minsk'] = "EuropeMinsk";
        TimezoneEnum[TimezoneEnum["EuropeMonaco"] = 'Europe/Monaco'] = "EuropeMonaco";
        TimezoneEnum[TimezoneEnum["EuropeMoscow"] = 'Europe/Moscow'] = "EuropeMoscow";
        TimezoneEnum[TimezoneEnum["EuropeOslo"] = 'Europe/Oslo'] = "EuropeOslo";
        TimezoneEnum[TimezoneEnum["EuropeParis"] = 'Europe/Paris'] = "EuropeParis";
        TimezoneEnum[TimezoneEnum["EuropePodgorica"] = 'Europe/Podgorica'] = "EuropePodgorica";
        TimezoneEnum[TimezoneEnum["EuropePrague"] = 'Europe/Prague'] = "EuropePrague";
        TimezoneEnum[TimezoneEnum["EuropeRiga"] = 'Europe/Riga'] = "EuropeRiga";
        TimezoneEnum[TimezoneEnum["EuropeRome"] = 'Europe/Rome'] = "EuropeRome";
        TimezoneEnum[TimezoneEnum["EuropeSamara"] = 'Europe/Samara'] = "EuropeSamara";
        TimezoneEnum[TimezoneEnum["EuropeSanMarino"] = 'Europe/San_Marino'] = "EuropeSanMarino";
        TimezoneEnum[TimezoneEnum["EuropeSarajevo"] = 'Europe/Sarajevo'] = "EuropeSarajevo";
        TimezoneEnum[TimezoneEnum["EuropeSaratov"] = 'Europe/Saratov'] = "EuropeSaratov";
        TimezoneEnum[TimezoneEnum["EuropeSimferopol"] = 'Europe/Simferopol'] = "EuropeSimferopol";
        TimezoneEnum[TimezoneEnum["EuropeSkopje"] = 'Europe/Skopje'] = "EuropeSkopje";
        TimezoneEnum[TimezoneEnum["EuropeSofia"] = 'Europe/Sofia'] = "EuropeSofia";
        TimezoneEnum[TimezoneEnum["EuropeStockholm"] = 'Europe/Stockholm'] = "EuropeStockholm";
        TimezoneEnum[TimezoneEnum["EuropeTallinn"] = 'Europe/Tallinn'] = "EuropeTallinn";
        TimezoneEnum[TimezoneEnum["EuropeTirane"] = 'Europe/Tirane'] = "EuropeTirane";
        TimezoneEnum[TimezoneEnum["EuropeUlyanovsk"] = 'Europe/Ulyanovsk'] = "EuropeUlyanovsk";
        TimezoneEnum[TimezoneEnum["EuropeVaduz"] = 'Europe/Vaduz'] = "EuropeVaduz";
        TimezoneEnum[TimezoneEnum["EuropeVatican"] = 'Europe/Vatican'] = "EuropeVatican";
        TimezoneEnum[TimezoneEnum["EuropeVienna"] = 'Europe/Vienna'] = "EuropeVienna";
        TimezoneEnum[TimezoneEnum["EuropeVilnius"] = 'Europe/Vilnius'] = "EuropeVilnius";
        TimezoneEnum[TimezoneEnum["EuropeVolgograd"] = 'Europe/Volgograd'] = "EuropeVolgograd";
        TimezoneEnum[TimezoneEnum["EuropeWarsaw"] = 'Europe/Warsaw'] = "EuropeWarsaw";
        TimezoneEnum[TimezoneEnum["EuropeZagreb"] = 'Europe/Zagreb'] = "EuropeZagreb";
        TimezoneEnum[TimezoneEnum["EuropeZurich"] = 'Europe/Zurich'] = "EuropeZurich";
        TimezoneEnum[TimezoneEnum["Gmt"] = 'GMT'] = "Gmt";
        TimezoneEnum[TimezoneEnum["IndianAntananarivo"] = 'Indian/Antananarivo'] = "IndianAntananarivo";
        TimezoneEnum[TimezoneEnum["IndianChagos"] = 'Indian/Chagos'] = "IndianChagos";
        TimezoneEnum[TimezoneEnum["IndianChristmas"] = 'Indian/Christmas'] = "IndianChristmas";
        TimezoneEnum[TimezoneEnum["IndianCocos"] = 'Indian/Cocos'] = "IndianCocos";
        TimezoneEnum[TimezoneEnum["IndianComoro"] = 'Indian/Comoro'] = "IndianComoro";
        TimezoneEnum[TimezoneEnum["IndianKerguelen"] = 'Indian/Kerguelen'] = "IndianKerguelen";
        TimezoneEnum[TimezoneEnum["IndianMahe"] = 'Indian/Mahe'] = "IndianMahe";
        TimezoneEnum[TimezoneEnum["IndianMaldives"] = 'Indian/Maldives'] = "IndianMaldives";
        TimezoneEnum[TimezoneEnum["IndianMauritius"] = 'Indian/Mauritius'] = "IndianMauritius";
        TimezoneEnum[TimezoneEnum["IndianMayotte"] = 'Indian/Mayotte'] = "IndianMayotte";
        TimezoneEnum[TimezoneEnum["IndianReunion"] = 'Indian/Reunion'] = "IndianReunion";
        TimezoneEnum[TimezoneEnum["PacificApia"] = 'Pacific/Apia'] = "PacificApia";
        TimezoneEnum[TimezoneEnum["PacificAuckland"] = 'Pacific/Auckland'] = "PacificAuckland";
        TimezoneEnum[TimezoneEnum["PacificBougainville"] = 'Pacific/Bougainville'] = "PacificBougainville";
        TimezoneEnum[TimezoneEnum["PacificChatham"] = 'Pacific/Chatham'] = "PacificChatham";
        TimezoneEnum[TimezoneEnum["PacificChuuk"] = 'Pacific/Chuuk'] = "PacificChuuk";
        TimezoneEnum[TimezoneEnum["PacificEaster"] = 'Pacific/Easter'] = "PacificEaster";
        TimezoneEnum[TimezoneEnum["PacificEfate"] = 'Pacific/Efate'] = "PacificEfate";
        TimezoneEnum[TimezoneEnum["PacificFakaofo"] = 'Pacific/Fakaofo'] = "PacificFakaofo";
        TimezoneEnum[TimezoneEnum["PacificFiji"] = 'Pacific/Fiji'] = "PacificFiji";
        TimezoneEnum[TimezoneEnum["PacificFunafuti"] = 'Pacific/Funafuti'] = "PacificFunafuti";
        TimezoneEnum[TimezoneEnum["PacificGalapagos"] = 'Pacific/Galapagos'] = "PacificGalapagos";
        TimezoneEnum[TimezoneEnum["PacificGambier"] = 'Pacific/Gambier'] = "PacificGambier";
        TimezoneEnum[TimezoneEnum["PacificGuadalcanal"] = 'Pacific/Guadalcanal'] = "PacificGuadalcanal";
        TimezoneEnum[TimezoneEnum["PacificGuam"] = 'Pacific/Guam'] = "PacificGuam";
        TimezoneEnum[TimezoneEnum["PacificHonolulu"] = 'Pacific/Honolulu'] = "PacificHonolulu";
        TimezoneEnum[TimezoneEnum["PacificKanton"] = 'Pacific/Kanton'] = "PacificKanton";
        TimezoneEnum[TimezoneEnum["PacificKiritimati"] = 'Pacific/Kiritimati'] = "PacificKiritimati";
        TimezoneEnum[TimezoneEnum["PacificKosrae"] = 'Pacific/Kosrae'] = "PacificKosrae";
        TimezoneEnum[TimezoneEnum["PacificKwajalein"] = 'Pacific/Kwajalein'] = "PacificKwajalein";
        TimezoneEnum[TimezoneEnum["PacificMajuro"] = 'Pacific/Majuro'] = "PacificMajuro";
        TimezoneEnum[TimezoneEnum["PacificMarquesas"] = 'Pacific/Marquesas'] = "PacificMarquesas";
        TimezoneEnum[TimezoneEnum["PacificMidway"] = 'Pacific/Midway'] = "PacificMidway";
        TimezoneEnum[TimezoneEnum["PacificNauru"] = 'Pacific/Nauru'] = "PacificNauru";
        TimezoneEnum[TimezoneEnum["PacificNiue"] = 'Pacific/Niue'] = "PacificNiue";
        TimezoneEnum[TimezoneEnum["PacificNorfolk"] = 'Pacific/Norfolk'] = "PacificNorfolk";
        TimezoneEnum[TimezoneEnum["PacificNoumea"] = 'Pacific/Noumea'] = "PacificNoumea";
        TimezoneEnum[TimezoneEnum["PacificPagoPago"] = 'Pacific/Pago_Pago'] = "PacificPagoPago";
        TimezoneEnum[TimezoneEnum["PacificPalau"] = 'Pacific/Palau'] = "PacificPalau";
        TimezoneEnum[TimezoneEnum["PacificPitcairn"] = 'Pacific/Pitcairn'] = "PacificPitcairn";
        TimezoneEnum[TimezoneEnum["PacificPohnpei"] = 'Pacific/Pohnpei'] = "PacificPohnpei";
        TimezoneEnum[TimezoneEnum["PacificPortMoresby"] = 'Pacific/Port_Moresby'] = "PacificPortMoresby";
        TimezoneEnum[TimezoneEnum["PacificRarotonga"] = 'Pacific/Rarotonga'] = "PacificRarotonga";
        TimezoneEnum[TimezoneEnum["PacificSaipan"] = 'Pacific/Saipan'] = "PacificSaipan";
        TimezoneEnum[TimezoneEnum["PacificTahiti"] = 'Pacific/Tahiti'] = "PacificTahiti";
        TimezoneEnum[TimezoneEnum["PacificTarawa"] = 'Pacific/Tarawa'] = "PacificTarawa";
        TimezoneEnum[TimezoneEnum["PacificTongatapu"] = 'Pacific/Tongatapu'] = "PacificTongatapu";
        TimezoneEnum[TimezoneEnum["PacificWake"] = 'Pacific/Wake'] = "PacificWake";
        TimezoneEnum[TimezoneEnum["PacificWallis"] = 'Pacific/Wallis'] = "PacificWallis";
        TimezoneEnum[TimezoneEnum["UsAlaska"] = 'US/Alaska'] = "UsAlaska";
        TimezoneEnum[TimezoneEnum["UsArizona"] = 'US/Arizona'] = "UsArizona";
        TimezoneEnum[TimezoneEnum["UsCentral"] = 'US/Central'] = "UsCentral";
        TimezoneEnum[TimezoneEnum["UsEastern"] = 'US/Eastern'] = "UsEastern";
        TimezoneEnum[TimezoneEnum["UsHawaii"] = 'US/Hawaii'] = "UsHawaii";
        TimezoneEnum[TimezoneEnum["UsMountain"] = 'US/Mountain'] = "UsMountain";
        TimezoneEnum[TimezoneEnum["UsPacific"] = 'US/Pacific'] = "UsPacific";
        TimezoneEnum[TimezoneEnum["Utc"] = 'UTC'] = "Utc";
        TimezoneEnum[TimezoneEnum["Profile"] = 'profile'] = "Profile";
    })(TimezoneEnum = CountdownDelayActionData.TimezoneEnum || (CountdownDelayActionData.TimezoneEnum = {}));
    let DelayUntilWeekdaysEnum;
    (function (DelayUntilWeekdaysEnum) {
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Friday"] = 'friday'] = "Friday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Monday"] = 'monday'] = "Monday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Saturday"] = 'saturday'] = "Saturday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Sunday"] = 'sunday'] = "Sunday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Thursday"] = 'thursday'] = "Thursday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Tuesday"] = 'tuesday'] = "Tuesday";
        DelayUntilWeekdaysEnum[DelayUntilWeekdaysEnum["Wednesday"] = 'wednesday'] = "Wednesday";
    })(DelayUntilWeekdaysEnum = CountdownDelayActionData.DelayUntilWeekdaysEnum || (CountdownDelayActionData.DelayUntilWeekdaysEnum = {}));
})(CountdownDelayActionData = exports.CountdownDelayActionData || (exports.CountdownDelayActionData = {}));
