// klaviyo-api/dist/model/couponCodeUpdateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CouponCodeUpdateQueryResourceObjectAttributes = void 0;
class CouponCodeUpdateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return CouponCodeUpdateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.CouponCodeUpdateQueryResourceObjectAttributes = CouponCodeUpdateQueryResourceObjectAttributes;
CouponCodeUpdateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "CouponCodeUpdateQueryResourceObjectAttributes.StatusEnum"
    },
    {
        "name": "expiresAt",
        "baseName": "expires_at",
        "type": "Date"
    }
];
(function (CouponCodeUpdateQueryResourceObjectAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AssignedToProfile"] = 'ASSIGNED_TO_PROFILE'] = "AssignedToProfile";
        StatusEnum[StatusEnum["Deleting"] = 'DELETING'] = "Deleting";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Unassigned"] = 'UNASSIGNED'] = "Unassigned";
        StatusEnum[StatusEnum["Used"] = 'USED'] = "Used";
        StatusEnum[StatusEnum["VersionNotActive"] = 'VERSION_NOT_ACTIVE'] = "VersionNotActive";
    })(StatusEnum = CouponCodeUpdateQueryResourceObjectAttributes.StatusEnum || (CouponCodeUpdateQueryResourceObjectAttributes.StatusEnum = {}));
})(CouponCodeUpdateQueryResourceObjectAttributes = exports.CouponCodeUpdateQueryResourceObjectAttributes || (exports.CouponCodeUpdateQueryResourceObjectAttributes = {}));
