// klaviyo-api/dist/model/couponCodeResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CouponCodeResponseObjectResourceAttributes = void 0;
class CouponCodeResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return CouponCodeResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.CouponCodeResponseObjectResourceAttributes = CouponCodeResponseObjectResourceAttributes;
CouponCodeResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "uniqueCode",
        "baseName": "unique_code",
        "type": "string"
    },
    {
        "name": "expiresAt",
        "baseName": "expires_at",
        "type": "Date"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CouponCodeResponseObjectResourceAttributes.StatusEnum"
    }
];
(function (CouponCodeResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AssignedToProfile"] = 'ASSIGNED_TO_PROFILE'] = "AssignedToProfile";
        StatusEnum[StatusEnum["Deleting"] = 'DELETING'] = "Deleting";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Unassigned"] = 'UNASSIGNED'] = "Unassigned";
        StatusEnum[StatusEnum["Used"] = 'USED'] = "Used";
        StatusEnum[StatusEnum["VersionNotActive"] = 'VERSION_NOT_ACTIVE'] = "VersionNotActive";
    })(StatusEnum = CouponCodeResponseObjectResourceAttributes.StatusEnum || (CouponCodeResponseObjectResourceAttributes.StatusEnum = {}));
})(CouponCodeResponseObjectResourceAttributes = exports.CouponCodeResponseObjectResourceAttributes || (exports.CouponCodeResponseObjectResourceAttributes = {}));
