// klaviyo-api/dist/model/customMetricDefinition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomMetricDefinition = void 0;
class CustomMetricDefinition {
    static getAttributeTypeMap() {
        return CustomMetricDefinition.attributeTypeMap;
    }
}
exports.CustomMetricDefinition = CustomMetricDefinition;
CustomMetricDefinition.attributeTypeMap = [
    {
        "name": "aggregationMethod",
        "baseName": "aggregation_method",
        "type": "CustomMetricDefinition.AggregationMethodEnum"
    },
    {
        "name": "metricGroups",
        "baseName": "metric_groups",
        "type": "Array<CustomMetricGroup>"
    }
];
(function (CustomMetricDefinition) {
    let AggregationMethodEnum;
    (function (AggregationMethodEnum) {
        AggregationMethodEnum[AggregationMethodEnum["Count"] = 'count'] = "Count";
        AggregationMethodEnum[AggregationMethodEnum["Value"] = 'value'] = "Value";
    })(AggregationMethodEnum = CustomMetricDefinition.AggregationMethodEnum || (CustomMetricDefinition.AggregationMethodEnum = {}));
})(CustomMetricDefinition = exports.CustomMetricDefinition || (exports.CustomMetricDefinition = {}));
