// klaviyo-api/dist/model/deviceMetadata.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceMetadata = void 0;
class DeviceMetadata {
    static getAttributeTypeMap() {
        return DeviceMetadata.attributeTypeMap;
    }
}
exports.DeviceMetadata = DeviceMetadata;
DeviceMetadata.attributeTypeMap = [
    {
        "name": "deviceId",
        "baseName": "device_id",
        "type": "string"
    },
    {
        "name": "klaviyoSdk",
        "baseName": "klaviyo_sdk",
        "type": "DeviceMetadata.KlaviyoSdkEnum"
    },
    {
        "name": "sdkVersion",
        "baseName": "sdk_version",
        "type": "string"
    },
    {
        "name": "deviceModel",
        "baseName": "device_model",
        "type": "string"
    },
    {
        "name": "osName",
        "baseName": "os_name",
        "type": "DeviceMetadata.OsNameEnum"
    },
    {
        "name": "osVersion",
        "baseName": "os_version",
        "type": "string"
    },
    {
        "name": "manufacturer",
        "baseName": "manufacturer",
        "type": "string"
    },
    {
        "name": "appName",
        "baseName": "app_name",
        "type": "string"
    },
    {
        "name": "appVersion",
        "baseName": "app_version",
        "type": "string"
    },
    {
        "name": "appBuild",
        "baseName": "app_build",
        "type": "string"
    },
    {
        "name": "appId",
        "baseName": "app_id",
        "type": "string"
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "DeviceMetadata.EnvironmentEnum"
    }
];
(function (DeviceMetadata) {
    let KlaviyoSdkEnum;
    (function (KlaviyoSdkEnum) {
        KlaviyoSdkEnum[KlaviyoSdkEnum["Android"] = 'android'] = "Android";
        KlaviyoSdkEnum[KlaviyoSdkEnum["FlutterCommunity"] = 'flutter_community'] = "FlutterCommunity";
        KlaviyoSdkEnum[KlaviyoSdkEnum["ReactNative"] = 'react_native'] = "ReactNative";
        KlaviyoSdkEnum[KlaviyoSdkEnum["Swift"] = 'swift'] = "Swift";
    })(KlaviyoSdkEnum = DeviceMetadata.KlaviyoSdkEnum || (DeviceMetadata.KlaviyoSdkEnum = {}));
    let OsNameEnum;
    (function (OsNameEnum) {
        OsNameEnum[OsNameEnum["Android"] = 'android'] = "Android";
        OsNameEnum[OsNameEnum["Ios"] = 'ios'] = "Ios";
        OsNameEnum[OsNameEnum["Ipados"] = 'ipados'] = "Ipados";
        OsNameEnum[OsNameEnum["Macos"] = 'macos'] = "Macos";
        OsNameEnum[OsNameEnum["Tvos"] = 'tvos'] = "Tvos";
    })(OsNameEnum = DeviceMetadata.OsNameEnum || (DeviceMetadata.OsNameEnum = {}));
    let EnvironmentEnum;
    (function (EnvironmentEnum) {
        EnvironmentEnum[EnvironmentEnum["Debug"] = 'debug'] = "Debug";
        EnvironmentEnum[EnvironmentEnum["Release"] = 'release'] = "Release";
    })(EnvironmentEnum = DeviceMetadata.EnvironmentEnum || (DeviceMetadata.EnvironmentEnum = {}));
})(DeviceMetadata = exports.DeviceMetadata || (exports.DeviceMetadata = {}));
