// klaviyo-api/dist/model/dynamicTrackingParam.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DynamicTrackingParam = void 0;
class DynamicTrackingParam {
    static getAttributeTypeMap() {
        return DynamicTrackingParam.attributeTypeMap;
    }
}
exports.DynamicTrackingParam = DynamicTrackingParam;
DynamicTrackingParam.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DynamicEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "DynamicTrackingParam.ValueEnum"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    }
];
(function (DynamicTrackingParam) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum[ValueEnum["CampaignId"] = 'campaign_id'] = "CampaignId";
        ValueEnum[ValueEnum["CampaignName"] = 'campaign_name'] = "CampaignName";
        ValueEnum[ValueEnum["CampaignNameId"] = 'campaign_name_id'] = "CampaignNameId";
        ValueEnum[ValueEnum["CampaignNameSendDay"] = 'campaign_name_send_day'] = "CampaignNameSendDay";
        ValueEnum[ValueEnum["EmailSubject"] = 'email_subject'] = "EmailSubject";
        ValueEnum[ValueEnum["GroupId"] = 'group_id'] = "GroupId";
        ValueEnum[ValueEnum["GroupName"] = 'group_name'] = "GroupName";
        ValueEnum[ValueEnum["GroupNameId"] = 'group_name_id'] = "GroupNameId";
        ValueEnum[ValueEnum["LinkAltText"] = 'link_alt_text'] = "LinkAltText";
        ValueEnum[ValueEnum["MessageType"] = 'message_type'] = "MessageType";
        ValueEnum[ValueEnum["ProfileExternalId"] = 'profile_external_id'] = "ProfileExternalId";
        ValueEnum[ValueEnum["ProfileId"] = 'profile_id'] = "ProfileId";
    })(ValueEnum = DynamicTrackingParam.ValueEnum || (DynamicTrackingParam.ValueEnum = {}));
})(DynamicTrackingParam = exports.DynamicTrackingParam || (exports.DynamicTrackingParam = {}));
