// klaviyo-api/dist/model/emailMarketingSuppression.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailMarketingSuppression = void 0;
class EmailMarketingSuppression {
    static getAttributeTypeMap() {
        return EmailMarketingSuppression.attributeTypeMap;
    }
}
exports.EmailMarketingSuppression = EmailMarketingSuppression;
EmailMarketingSuppression.attributeTypeMap = [
    {
        "name": "reason",
        "baseName": "reason",
        "type": "EmailMarketingSuppression.ReasonEnum"
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date"
    }
];
(function (EmailMarketingSuppression) {
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum[ReasonEnum["HardBounce"] = 'HARD_BOUNCE'] = "HardBounce";
        ReasonEnum[ReasonEnum["InvalidEmail"] = 'INVALID_EMAIL'] = "InvalidEmail";
        ReasonEnum[ReasonEnum["SpamComplaint"] = 'SPAM_COMPLAINT'] = "SpamComplaint";
        ReasonEnum[ReasonEnum["Unsubscribe"] = 'UNSUBSCRIBE'] = "Unsubscribe";
        ReasonEnum[ReasonEnum["UserSuppressed"] = 'USER_SUPPRESSED'] = "UserSuppressed";
    })(ReasonEnum = EmailMarketingSuppression.ReasonEnum || (EmailMarketingSuppression.ReasonEnum = {}));
})(EmailMarketingSuppression = exports.EmailMarketingSuppression || (exports.EmailMarketingSuppression = {}));
