// klaviyo-api/dist/model/existenceOperatorFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExistenceOperatorFilter = void 0;
class ExistenceOperatorFilter {
    static getAttributeTypeMap() {
        return ExistenceOperatorFilter.attributeTypeMap;
    }
}
exports.ExistenceOperatorFilter = ExistenceOperatorFilter;
ExistenceOperatorFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ExistenceEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ExistenceOperatorFilter.OperatorEnum"
    }
];
(function (ExistenceOperatorFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["IsSet"] = 'is-set'] = "IsSet";
        OperatorEnum[OperatorEnum["NotSet"] = 'not-set'] = "NotSet";
    })(OperatorEnum = ExistenceOperatorFilter.OperatorEnum || (ExistenceOperatorFilter.OperatorEnum = {}));
})(ExistenceOperatorFilter = exports.ExistenceOperatorFilter || (exports.ExistenceOperatorFilter = {}));
