// klaviyo-api/dist/model/flowPushNotification.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowPushNotification = void 0;
class FlowPushNotification {
    constructor() {
        this['sound'] = false;
        this['badge'] = false;
        /**
        * See PushLinkAction in app.  This is not a flow action, but the literal action that should be taken when the push notification is tapped.
        */
        this['onOpen'] = FlowPushNotification.OnOpenEnum.Home;
        this['smartSendingEnabled'] = true;
    }
    static getAttributeTypeMap() {
        return FlowPushNotification.attributeTypeMap;
    }
}
exports.FlowPushNotification = FlowPushNotification;
FlowPushNotification.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "body",
        "baseName": "body",
        "type": "string"
    },
    {
        "name": "sound",
        "baseName": "sound",
        "type": "boolean"
    },
    {
        "name": "badge",
        "baseName": "badge",
        "type": "boolean"
    },
    {
        "name": "imageId",
        "baseName": "image_id",
        "type": "string"
    },
    {
        "name": "dynamicImage",
        "baseName": "dynamic_image",
        "type": "string"
    },
    {
        "name": "onOpen",
        "baseName": "on_open",
        "type": "FlowPushNotification.OnOpenEnum"
    },
    {
        "name": "iosLink",
        "baseName": "ios_link",
        "type": "string"
    },
    {
        "name": "androidLink",
        "baseName": "android_link",
        "type": "string"
    },
    {
        "name": "conversionMetricId",
        "baseName": "conversion_metric_id",
        "type": "string"
    },
    {
        "name": "smartSendingEnabled",
        "baseName": "smart_sending_enabled",
        "type": "boolean"
    },
    {
        "name": "additionalFilters",
        "baseName": "additional_filters",
        "type": "FlowPushNotificationAdditionalFilters"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (FlowPushNotification) {
    let OnOpenEnum;
    (function (OnOpenEnum) {
        OnOpenEnum[OnOpenEnum["Home"] = 'home'] = "Home";
        OnOpenEnum[OnOpenEnum["Link"] = 'link'] = "Link";
    })(OnOpenEnum = FlowPushNotification.OnOpenEnum || (FlowPushNotification.OnOpenEnum = {}));
})(FlowPushNotification = exports.FlowPushNotification || (exports.FlowPushNotification = {}));
