// klaviyo-api/dist/model/flowTrackingSettingDynamicParam.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowTrackingSettingDynamicParam = void 0;
class FlowTrackingSettingDynamicParam {
    static getAttributeTypeMap() {
        return FlowTrackingSettingDynamicParam.attributeTypeMap;
    }
}
exports.FlowTrackingSettingDynamicParam = FlowTrackingSettingDynamicParam;
FlowTrackingSettingDynamicParam.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DynamicEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "FlowTrackingSettingDynamicParam.ValueEnum"
    }
];
(function (FlowTrackingSettingDynamicParam) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum[ValueEnum["EmailSubject"] = 'email_subject'] = "EmailSubject";
        ValueEnum[ValueEnum["FlowId"] = 'flow_id'] = "FlowId";
        ValueEnum[ValueEnum["FlowName"] = 'flow_name'] = "FlowName";
        ValueEnum[ValueEnum["LinkAltText"] = 'link_alt_text'] = "LinkAltText";
        ValueEnum[ValueEnum["MessageName"] = 'message_name'] = "MessageName";
        ValueEnum[ValueEnum["MessageNameId"] = 'message_name_id'] = "MessageNameId";
        ValueEnum[ValueEnum["MessageType"] = 'message_type'] = "MessageType";
        ValueEnum[ValueEnum["ProfileExternalId"] = 'profile_external_id'] = "ProfileExternalId";
        ValueEnum[ValueEnum["ProfileId"] = 'profile_id'] = "ProfileId";
    })(ValueEnum = FlowTrackingSettingDynamicParam.ValueEnum || (FlowTrackingSettingDynamicParam.ValueEnum = {}));
})(FlowTrackingSettingDynamicParam = exports.FlowTrackingSettingDynamicParam || (exports.FlowTrackingSettingDynamicParam = {}));
