// klaviyo-api/dist/model/flowsProfileMetricCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowsProfileMetricCondition = void 0;
class FlowsProfileMetricCondition {
    static getAttributeTypeMap() {
        return FlowsProfileMetricCondition.attributeTypeMap;
    }
}
exports.FlowsProfileMetricCondition = FlowsProfileMetricCondition;
FlowsProfileMetricCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfileMetricEnum"
    },
    {
        "name": "metricId",
        "baseName": "metric_id",
        "type": "string"
    },
    {
        "name": "measurement",
        "baseName": "measurement",
        "type": "FlowsProfileMetricCondition.MeasurementEnum"
    },
    {
        "name": "measurementFilter",
        "baseName": "measurement_filter",
        "type": "NumericOperatorFilter"
    },
    {
        "name": "timeframeFilter",
        "baseName": "timeframe_filter",
        "type": "FlowsProfileMetricConditionTimeframeFilter"
    },
    {
        "name": "metricFilters",
        "baseName": "metric_filters",
        "type": "Array<ProfileMetricPropertyFilter>"
    }
];
(function (FlowsProfileMetricCondition) {
    let MeasurementEnum;
    (function (MeasurementEnum) {
        MeasurementEnum[MeasurementEnum["Count"] = 'count'] = "Count";
        MeasurementEnum[MeasurementEnum["Sum"] = 'sum'] = "Sum";
    })(MeasurementEnum = FlowsProfileMetricCondition.MeasurementEnum || (FlowsProfileMetricCondition.MeasurementEnum = {}));
})(FlowsProfileMetricCondition = exports.FlowsProfileMetricCondition || (exports.FlowsProfileMetricCondition = {}));
