// klaviyo-api/dist/model/flowV2ResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowV2ResponseObjectResourceAttributes = void 0;
class FlowV2ResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return FlowV2ResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.FlowV2ResponseObjectResourceAttributes = FlowV2ResponseObjectResourceAttributes;
FlowV2ResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "string"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "triggerType",
        "baseName": "trigger_type",
        "type": "FlowV2ResponseObjectResourceAttributes.TriggerTypeEnum"
    }
];
(function (FlowV2ResponseObjectResourceAttributes) {
    let TriggerTypeEnum;
    (function (TriggerTypeEnum) {
        TriggerTypeEnum[TriggerTypeEnum["AddedToList"] = 'Added to List'] = "AddedToList";
        TriggerTypeEnum[TriggerTypeEnum["DateBased"] = 'Date Based'] = "DateBased";
        TriggerTypeEnum[TriggerTypeEnum["LowInventory"] = 'Low Inventory'] = "LowInventory";
        TriggerTypeEnum[TriggerTypeEnum["Metric"] = 'Metric'] = "Metric";
        TriggerTypeEnum[TriggerTypeEnum["PriceDrop"] = 'Price Drop'] = "PriceDrop";
        TriggerTypeEnum[TriggerTypeEnum["Unconfigured"] = 'Unconfigured'] = "Unconfigured";
    })(TriggerTypeEnum = FlowV2ResponseObjectResourceAttributes.TriggerTypeEnum || (FlowV2ResponseObjectResourceAttributes.TriggerTypeEnum = {}));
})(FlowV2ResponseObjectResourceAttributes = exports.FlowV2ResponseObjectResourceAttributes || (exports.FlowV2ResponseObjectResourceAttributes = {}));
