// klaviyo-api/dist/model/flowSms.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowSms = void 0;
class FlowSms {
    constructor() {
        this['shortenLinks'] = true;
        this['includeContactCard'] = false;
        this['addOrgPrefix'] = true;
        this['addInfoLink'] = true;
        this['addOptOutLanguage'] = false;
        this['smartSendingEnabled'] = true;
        this['smsQuietHoursEnabled'] = true;
        this['transactional'] = false;
        this['addTrackingParams'] = false;
    }
    static getAttributeTypeMap() {
        return FlowSms.attributeTypeMap;
    }
}
exports.FlowSms = FlowSms;
FlowSms.attributeTypeMap = [
    {
        "name": "body",
        "baseName": "body",
        "type": "string"
    },
    {
        "name": "imageId",
        "baseName": "image_id",
        "type": "string"
    },
    {
        "name": "dynamicImage",
        "baseName": "dynamic_image",
        "type": "string"
    },
    {
        "name": "shortenLinks",
        "baseName": "shorten_links",
        "type": "boolean"
    },
    {
        "name": "includeContactCard",
        "baseName": "include_contact_card",
        "type": "boolean"
    },
    {
        "name": "addOrgPrefix",
        "baseName": "add_org_prefix",
        "type": "boolean"
    },
    {
        "name": "addInfoLink",
        "baseName": "add_info_link",
        "type": "boolean"
    },
    {
        "name": "addOptOutLanguage",
        "baseName": "add_opt_out_language",
        "type": "boolean"
    },
    {
        "name": "smartSendingEnabled",
        "baseName": "smart_sending_enabled",
        "type": "boolean"
    },
    {
        "name": "smsQuietHoursEnabled",
        "baseName": "sms_quiet_hours_enabled",
        "type": "boolean"
    },
    {
        "name": "transactional",
        "baseName": "transactional",
        "type": "boolean"
    },
    {
        "name": "addTrackingParams",
        "baseName": "add_tracking_params",
        "type": "boolean"
    },
    {
        "name": "customTrackingParams",
        "baseName": "custom_tracking_params",
        "type": "Array<UtmParam>"
    },
    {
        "name": "additionalFilters",
        "baseName": "additional_filters",
        "type": "FlowSmsAdditionalFilters"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
