// klaviyo-api/dist/model/flowEmail.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowEmail = void 0;
class FlowEmail {
    constructor() {
        this['smartSendingEnabled'] = true;
        this['transactional'] = false;
        this['addTrackingParams'] = false;
    }
    static getAttributeTypeMap() {
        return FlowEmail.attributeTypeMap;
    }
}
exports.FlowEmail = FlowEmail;
FlowEmail.attributeTypeMap = [
    {
        "name": "fromEmail",
        "baseName": "from_email",
        "type": "string"
    },
    {
        "name": "fromLabel",
        "baseName": "from_label",
        "type": "string"
    },
    {
        "name": "replyToEmail",
        "baseName": "reply_to_email",
        "type": "string"
    },
    {
        "name": "ccEmail",
        "baseName": "cc_email",
        "type": "string"
    },
    {
        "name": "bccEmail",
        "baseName": "bcc_email",
        "type": "string"
    },
    {
        "name": "subjectLine",
        "baseName": "subject_line",
        "type": "string"
    },
    {
        "name": "previewText",
        "baseName": "preview_text",
        "type": "string"
    },
    {
        "name": "templateId",
        "baseName": "template_id",
        "type": "string"
    },
    {
        "name": "smartSendingEnabled",
        "baseName": "smart_sending_enabled",
        "type": "boolean"
    },
    {
        "name": "transactional",
        "baseName": "transactional",
        "type": "boolean"
    },
    {
        "name": "addTrackingParams",
        "baseName": "add_tracking_params",
        "type": "boolean"
    },
    {
        "name": "customTrackingParams",
        "baseName": "custom_tracking_params",
        "type": "Array<UtmParam>"
    },
    {
        "name": "additionalFilters",
        "baseName": "additional_filters",
        "type": "FlowEmailAdditionalFilters"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
