// klaviyo-api/dist/model/flowSeriesRequestDTOResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowSeriesRequestDTOResourceObjectAttributes = void 0;
class FlowSeriesRequestDTOResourceObjectAttributes {
    static getAttributeTypeMap() {
        return FlowSeriesRequestDTOResourceObjectAttributes.attributeTypeMap;
    }
}
exports.FlowSeriesRequestDTOResourceObjectAttributes = FlowSeriesRequestDTOResourceObjectAttributes;
FlowSeriesRequestDTOResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "statistics",
        "baseName": "statistics",
        "type": "Array<FlowSeriesRequestDTOResourceObjectAttributes.StatisticsEnum>"
    },
    {
        "name": "timeframe",
        "baseName": "timeframe",
        "type": "CampaignValuesRequestDTOResourceObjectAttributesTimeframe"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "FlowSeriesRequestDTOResourceObjectAttributes.IntervalEnum"
    },
    {
        "name": "conversionMetricId",
        "baseName": "conversion_metric_id",
        "type": "string"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "string"
    }
];
(function (FlowSeriesRequestDTOResourceObjectAttributes) {
    let StatisticsEnum;
    (function (StatisticsEnum) {
        StatisticsEnum[StatisticsEnum["AverageOrderValue"] = 'average_order_value'] = "AverageOrderValue";
        StatisticsEnum[StatisticsEnum["BounceRate"] = 'bounce_rate'] = "BounceRate";
        StatisticsEnum[StatisticsEnum["Bounced"] = 'bounced'] = "Bounced";
        StatisticsEnum[StatisticsEnum["BouncedOrFailed"] = 'bounced_or_failed'] = "BouncedOrFailed";
        StatisticsEnum[StatisticsEnum["BouncedOrFailedRate"] = 'bounced_or_failed_rate'] = "BouncedOrFailedRate";
        StatisticsEnum[StatisticsEnum["ClickRate"] = 'click_rate'] = "ClickRate";
        StatisticsEnum[StatisticsEnum["ClickToOpenRate"] = 'click_to_open_rate'] = "ClickToOpenRate";
        StatisticsEnum[StatisticsEnum["Clicks"] = 'clicks'] = "Clicks";
        StatisticsEnum[StatisticsEnum["ClicksUnique"] = 'clicks_unique'] = "ClicksUnique";
        StatisticsEnum[StatisticsEnum["ConversionRate"] = 'conversion_rate'] = "ConversionRate";
        StatisticsEnum[StatisticsEnum["ConversionUniques"] = 'conversion_uniques'] = "ConversionUniques";
        StatisticsEnum[StatisticsEnum["ConversionValue"] = 'conversion_value'] = "ConversionValue";
        StatisticsEnum[StatisticsEnum["Conversions"] = 'conversions'] = "Conversions";
        StatisticsEnum[StatisticsEnum["Delivered"] = 'delivered'] = "Delivered";
        StatisticsEnum[StatisticsEnum["DeliveryRate"] = 'delivery_rate'] = "DeliveryRate";
        StatisticsEnum[StatisticsEnum["Failed"] = 'failed'] = "Failed";
        StatisticsEnum[StatisticsEnum["FailedRate"] = 'failed_rate'] = "FailedRate";
        StatisticsEnum[StatisticsEnum["OpenRate"] = 'open_rate'] = "OpenRate";
        StatisticsEnum[StatisticsEnum["Opens"] = 'opens'] = "Opens";
        StatisticsEnum[StatisticsEnum["OpensUnique"] = 'opens_unique'] = "OpensUnique";
        StatisticsEnum[StatisticsEnum["Recipients"] = 'recipients'] = "Recipients";
        StatisticsEnum[StatisticsEnum["RevenuePerRecipient"] = 'revenue_per_recipient'] = "RevenuePerRecipient";
        StatisticsEnum[StatisticsEnum["SpamComplaintRate"] = 'spam_complaint_rate'] = "SpamComplaintRate";
        StatisticsEnum[StatisticsEnum["SpamComplaints"] = 'spam_complaints'] = "SpamComplaints";
        StatisticsEnum[StatisticsEnum["UnsubscribeRate"] = 'unsubscribe_rate'] = "UnsubscribeRate";
        StatisticsEnum[StatisticsEnum["UnsubscribeUniques"] = 'unsubscribe_uniques'] = "UnsubscribeUniques";
        StatisticsEnum[StatisticsEnum["Unsubscribes"] = 'unsubscribes'] = "Unsubscribes";
    })(StatisticsEnum = FlowSeriesRequestDTOResourceObjectAttributes.StatisticsEnum || (FlowSeriesRequestDTOResourceObjectAttributes.StatisticsEnum = {}));
    let IntervalEnum;
    (function (IntervalEnum) {
        IntervalEnum[IntervalEnum["Daily"] = 'daily'] = "Daily";
        IntervalEnum[IntervalEnum["Hourly"] = 'hourly'] = "Hourly";
        IntervalEnum[IntervalEnum["Monthly"] = 'monthly'] = "Monthly";
        IntervalEnum[IntervalEnum["Weekly"] = 'weekly'] = "Weekly";
    })(IntervalEnum = FlowSeriesRequestDTOResourceObjectAttributes.IntervalEnum || (FlowSeriesRequestDTOResourceObjectAttributes.IntervalEnum = {}));
})(FlowSeriesRequestDTOResourceObjectAttributes = exports.FlowSeriesRequestDTOResourceObjectAttributes || (exports.FlowSeriesRequestDTOResourceObjectAttributes = {}));
