// klaviyo-api/dist/model/flowResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowResponseObjectResourceAttributes = void 0;
class FlowResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return FlowResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.FlowResponseObjectResourceAttributes = FlowResponseObjectResourceAttributes;
FlowResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "string"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "triggerType",
        "baseName": "trigger_type",
        "type": "FlowResponseObjectResourceAttributes.TriggerTypeEnum"
    }
];
(function (FlowResponseObjectResourceAttributes) {
    let TriggerTypeEnum;
    (function (TriggerTypeEnum) {
        TriggerTypeEnum[TriggerTypeEnum["AddedToList"] = 'Added to List'] = "AddedToList";
        TriggerTypeEnum[TriggerTypeEnum["DateBased"] = 'Date Based'] = "DateBased";
        TriggerTypeEnum[TriggerTypeEnum["LowInventory"] = 'Low Inventory'] = "LowInventory";
        TriggerTypeEnum[TriggerTypeEnum["Metric"] = 'Metric'] = "Metric";
        TriggerTypeEnum[TriggerTypeEnum["PriceDrop"] = 'Price Drop'] = "PriceDrop";
        TriggerTypeEnum[TriggerTypeEnum["Unconfigured"] = 'Unconfigured'] = "Unconfigured";
    })(TriggerTypeEnum = FlowResponseObjectResourceAttributes.TriggerTypeEnum || (FlowResponseObjectResourceAttributes.TriggerTypeEnum = {}));
})(FlowResponseObjectResourceAttributes = exports.FlowResponseObjectResourceAttributes || (exports.FlowResponseObjectResourceAttributes = {}));
