// klaviyo-api/dist/model/formVersionResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormVersionResponseObjectResourceAttributes = void 0;
class FormVersionResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return FormVersionResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.FormVersionResponseObjectResourceAttributes = FormVersionResponseObjectResourceAttributes;
FormVersionResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "formType",
        "baseName": "form_type",
        "type": "FormVersionResponseObjectResourceAttributes.FormTypeEnum"
    },
    {
        "name": "abTest",
        "baseName": "ab_test",
        "type": "FormVersionABTest"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "FormVersionResponseObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "createdAt",
        "baseName": "created_at",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updated_at",
        "type": "Date"
    }
];
(function (FormVersionResponseObjectResourceAttributes) {
    let FormTypeEnum;
    (function (FormTypeEnum) {
        FormTypeEnum[FormTypeEnum["Banner"] = 'banner'] = "Banner";
        FormTypeEnum[FormTypeEnum["Embed"] = 'embed'] = "Embed";
        FormTypeEnum[FormTypeEnum["Flyout"] = 'flyout'] = "Flyout";
        FormTypeEnum[FormTypeEnum["FullPage"] = 'full_page'] = "FullPage";
        FormTypeEnum[FormTypeEnum["Popup"] = 'popup'] = "Popup";
    })(FormTypeEnum = FormVersionResponseObjectResourceAttributes.FormTypeEnum || (FormVersionResponseObjectResourceAttributes.FormTypeEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
    })(StatusEnum = FormVersionResponseObjectResourceAttributes.StatusEnum || (FormVersionResponseObjectResourceAttributes.StatusEnum = {}));
})(FormVersionResponseObjectResourceAttributes = exports.FormVersionResponseObjectResourceAttributes || (exports.FormVersionResponseObjectResourceAttributes = {}));
