// klaviyo-api/dist/model/formResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormResponseObjectResourceAttributes = void 0;
class FormResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return FormResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.FormResponseObjectResourceAttributes = FormResponseObjectResourceAttributes;
FormResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "FormResponseObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "abTest",
        "baseName": "ab_test",
        "type": "boolean"
    },
    {
        "name": "createdAt",
        "baseName": "created_at",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updated_at",
        "type": "Date"
    }
];
(function (FormResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
    })(StatusEnum = FormResponseObjectResourceAttributes.StatusEnum || (FormResponseObjectResourceAttributes.StatusEnum = {}));
})(FormResponseObjectResourceAttributes = exports.FormResponseObjectResourceAttributes || (exports.FormResponseObjectResourceAttributes = {}));
