// klaviyo-api/dist/model/formValuesRequestDTOResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormValuesRequestDTOResourceObjectAttributes = void 0;
class FormValuesRequestDTOResourceObjectAttributes {
    static getAttributeTypeMap() {
        return FormValuesRequestDTOResourceObjectAttributes.attributeTypeMap;
    }
}
exports.FormValuesRequestDTOResourceObjectAttributes = FormValuesRequestDTOResourceObjectAttributes;
FormValuesRequestDTOResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "statistics",
        "baseName": "statistics",
        "type": "Array<FormValuesRequestDTOResourceObjectAttributes.StatisticsEnum>"
    },
    {
        "name": "timeframe",
        "baseName": "timeframe",
        "type": "CampaignValuesRequestDTOResourceObjectAttributesTimeframe"
    },
    {
        "name": "groupBy",
        "baseName": "group_by",
        "type": "Array<FormValuesRequestDTOResourceObjectAttributes.GroupByEnum>"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "string"
    }
];
(function (FormValuesRequestDTOResourceObjectAttributes) {
    let StatisticsEnum;
    (function (StatisticsEnum) {
        StatisticsEnum[StatisticsEnum["ClosedForm"] = 'closed_form'] = "ClosedForm";
        StatisticsEnum[StatisticsEnum["ClosedFormUniques"] = 'closed_form_uniques'] = "ClosedFormUniques";
        StatisticsEnum[StatisticsEnum["QualifiedForm"] = 'qualified_form'] = "QualifiedForm";
        StatisticsEnum[StatisticsEnum["QualifiedFormUniques"] = 'qualified_form_uniques'] = "QualifiedFormUniques";
        StatisticsEnum[StatisticsEnum["SubmitRate"] = 'submit_rate'] = "SubmitRate";
        StatisticsEnum[StatisticsEnum["Submits"] = 'submits'] = "Submits";
        StatisticsEnum[StatisticsEnum["SubmittedFormStep"] = 'submitted_form_step'] = "SubmittedFormStep";
        StatisticsEnum[StatisticsEnum["SubmittedFormStepUniques"] = 'submitted_form_step_uniques'] = "SubmittedFormStepUniques";
        StatisticsEnum[StatisticsEnum["ViewedForm"] = 'viewed_form'] = "ViewedForm";
        StatisticsEnum[StatisticsEnum["ViewedFormStep"] = 'viewed_form_step'] = "ViewedFormStep";
        StatisticsEnum[StatisticsEnum["ViewedFormStepUniques"] = 'viewed_form_step_uniques'] = "ViewedFormStepUniques";
        StatisticsEnum[StatisticsEnum["ViewedFormUniques"] = 'viewed_form_uniques'] = "ViewedFormUniques";
    })(StatisticsEnum = FormValuesRequestDTOResourceObjectAttributes.StatisticsEnum || (FormValuesRequestDTOResourceObjectAttributes.StatisticsEnum = {}));
    let GroupByEnum;
    (function (GroupByEnum) {
        GroupByEnum[GroupByEnum["Id"] = 'form_id'] = "Id";
        GroupByEnum[GroupByEnum["VersionId"] = 'form_version_id'] = "VersionId";
    })(GroupByEnum = FormValuesRequestDTOResourceObjectAttributes.GroupByEnum || (FormValuesRequestDTOResourceObjectAttributes.GroupByEnum = {}));
})(FormValuesRequestDTOResourceObjectAttributes = exports.FormValuesRequestDTOResourceObjectAttributes || (exports.FormValuesRequestDTOResourceObjectAttributes = {}));
