// klaviyo-api/dist/model/inTheLastBaseRelativeDateFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InTheLastBaseRelativeDateFilter = void 0;
class InTheLastBaseRelativeDateFilter {
    static getAttributeTypeMap() {
        return InTheLastBaseRelativeDateFilter.attributeTypeMap;
    }
}
exports.InTheLastBaseRelativeDateFilter = InTheLastBaseRelativeDateFilter;
InTheLastBaseRelativeDateFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "InTheLastEnum"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "InTheLastBaseRelativeDateFilter.UnitEnum"
    },
    {
        "name": "quantity",
        "baseName": "quantity",
        "type": "number"
    }
];
(function (InTheLastBaseRelativeDateFilter) {
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Day"] = 'day'] = "Day";
        UnitEnum[UnitEnum["Hour"] = 'hour'] = "Hour";
        UnitEnum[UnitEnum["Week"] = 'week'] = "Week";
    })(UnitEnum = InTheLastBaseRelativeDateFilter.UnitEnum || (InTheLastBaseRelativeDateFilter.UnitEnum = {}));
})(InTheLastBaseRelativeDateFilter = exports.InTheLastBaseRelativeDateFilter || (exports.InTheLastBaseRelativeDateFilter = {}));
