// klaviyo-api/dist/model/listSubstringFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSubstringFilter = void 0;
class ListSubstringFilter {
    static getAttributeTypeMap() {
        return ListSubstringFilter.attributeTypeMap;
    }
}
exports.ListSubstringFilter = ListSubstringFilter;
ListSubstringFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ListEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ListSubstringFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (ListSubstringFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["ContainsSubstring"] = 'contains-substring'] = "ContainsSubstring";
        OperatorEnum[OperatorEnum["NotContainsSubstring"] = 'not-contains-substring'] = "NotContainsSubstring";
    })(OperatorEnum = ListSubstringFilter.OperatorEnum || (ListSubstringFilter.OperatorEnum = {}));
})(ListSubstringFilter = exports.ListSubstringFilter || (exports.ListSubstringFilter = {}));
