// klaviyo-api/dist/model/listUpdateActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUpdateActionData = void 0;
class ListUpdateActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = ListUpdateActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return ListUpdateActionData.attributeTypeMap;
    }
}
exports.ListUpdateActionData = ListUpdateActionData;
ListUpdateActionData.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "onExecution",
        "baseName": "on_execution",
        "type": "boolean"
    },
    {
        "name": "listId",
        "baseName": "list_id",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "ListUpdateActionData.StatusEnum"
    }
];
(function (ListUpdateActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = ListUpdateActionData.StatusEnum || (ListUpdateActionData.StatusEnum = {}));
})(ListUpdateActionData = exports.ListUpdateActionData || (exports.ListUpdateActionData = {}));
