// klaviyo-api/dist/model/listRetrieveResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRetrieveResponseObjectResourceAttributes = void 0;
class ListRetrieveResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return ListRetrieveResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.ListRetrieveResponseObjectResourceAttributes = ListRetrieveResponseObjectResourceAttributes;
ListRetrieveResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "optInProcess",
        "baseName": "opt_in_process",
        "type": "ListRetrieveResponseObjectResourceAttributes.OptInProcessEnum"
    }
];
(function (ListRetrieveResponseObjectResourceAttributes) {
    let OptInProcessEnum;
    (function (OptInProcessEnum) {
        OptInProcessEnum[OptInProcessEnum["DoubleOptIn"] = 'double_opt_in'] = "DoubleOptIn";
        OptInProcessEnum[OptInProcessEnum["SingleOptIn"] = 'single_opt_in'] = "SingleOptIn";
    })(OptInProcessEnum = ListRetrieveResponseObjectResourceAttributes.OptInProcessEnum || (ListRetrieveResponseObjectResourceAttributes.OptInProcessEnum = {}));
})(ListRetrieveResponseObjectResourceAttributes = exports.ListRetrieveResponseObjectResourceAttributes || (exports.ListRetrieveResponseObjectResourceAttributes = {}));
