// klaviyo-api/dist/model/listLengthFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListLengthFilter = void 0;
class ListLengthFilter {
    static getAttributeTypeMap() {
        return ListLengthFilter.attributeTypeMap;
    }
}
exports.ListLengthFilter = ListLengthFilter;
ListLengthFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ListEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ListLengthFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (ListLengthFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Equals"] = 'length-equals'] = "Equals";
        OperatorEnum[OperatorEnum["GreaterThan"] = 'length-greater-than'] = "GreaterThan";
        OperatorEnum[OperatorEnum["GreaterThanOrEqual"] = 'length-greater-than-or-equal'] = "GreaterThanOrEqual";
        OperatorEnum[OperatorEnum["LessThan"] = 'length-less-than'] = "LessThan";
        OperatorEnum[OperatorEnum["LessThanOrEqual"] = 'length-less-than-or-equal'] = "LessThanOrEqual";
    })(OperatorEnum = ListLengthFilter.OperatorEnum || (ListLengthFilter.OperatorEnum = {}));
})(ListLengthFilter = exports.ListLengthFilter || (exports.ListLengthFilter = {}));
