// klaviyo-api/dist/model/listSetFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSetFilter = void 0;
class ListSetFilter {
    static getAttributeTypeMap() {
        return ListSetFilter.attributeTypeMap;
    }
}
exports.ListSetFilter = ListSetFilter;
ListSetFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ListEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ListSetFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<string>"
    }
];
(function (ListSetFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["ContainsAll"] = 'contains-all'] = "ContainsAll";
        OperatorEnum[OperatorEnum["ContainsAny"] = 'contains-any'] = "ContainsAny";
        OperatorEnum[OperatorEnum["NotContainsAll"] = 'not-contains-all'] = "NotContainsAll";
        OperatorEnum[OperatorEnum["NotContainsAny"] = 'not-contains-any'] = "NotContainsAny";
    })(OperatorEnum = ListSetFilter.OperatorEnum || (ListSetFilter.OperatorEnum = {}));
})(ListSetFilter = exports.ListSetFilter || (exports.ListSetFilter = {}));
