// klaviyo-api/dist/model/listRegexOperatorContainsFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRegexOperatorContainsFilter = void 0;
class ListRegexOperatorContainsFilter {
    static getAttributeTypeMap() {
        return ListRegexOperatorContainsFilter.attributeTypeMap;
    }
}
exports.ListRegexOperatorContainsFilter = ListRegexOperatorContainsFilter;
ListRegexOperatorContainsFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ListEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ListRegexOperatorContainsFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (ListRegexOperatorContainsFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["ContainsEndsWith"] = 'contains-ends-with'] = "ContainsEndsWith";
        OperatorEnum[OperatorEnum["ContainsStartsWith"] = 'contains-starts-with'] = "ContainsStartsWith";
        OperatorEnum[OperatorEnum["NotContainsEndsWith"] = 'not-contains-ends-with'] = "NotContainsEndsWith";
        OperatorEnum[OperatorEnum["NotContainsStartsWith"] = 'not-contains-starts-with'] = "NotContainsStartsWith";
    })(OperatorEnum = ListRegexOperatorContainsFilter.OperatorEnum || (ListRegexOperatorContainsFilter.OperatorEnum = {}));
})(ListRegexOperatorContainsFilter = exports.ListRegexOperatorContainsFilter || (exports.ListRegexOperatorContainsFilter = {}));
