// klaviyo-api/dist/model/listContainsOperatorFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListContainsOperatorFilter = void 0;
class ListContainsOperatorFilter {
    static getAttributeTypeMap() {
        return ListContainsOperatorFilter.attributeTypeMap;
    }
}
exports.ListContainsOperatorFilter = ListContainsOperatorFilter;
ListContainsOperatorFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ListEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ListContainsOperatorFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (ListContainsOperatorFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Contains"] = 'contains'] = "Contains";
        OperatorEnum[OperatorEnum["NotContains"] = 'not-contains'] = "NotContains";
    })(OperatorEnum = ListContainsOperatorFilter.OperatorEnum || (ListContainsOperatorFilter.OperatorEnum = {}));
})(ListContainsOperatorFilter = exports.ListContainsOperatorFilter || (exports.ListContainsOperatorFilter = {}));
