// klaviyo-api/dist/model/localStaticSend.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalStaticSend = void 0;
class LocalStaticSend {
    constructor() {
        /**
        * Determines if we should send to local recipient timezone if the given time has passed. Only applicable to local sends.
        */
        this['sendPastRecipientsImmediately'] = false;
    }
    static getAttributeTypeMap() {
        return LocalStaticSend.attributeTypeMap;
    }
}
exports.LocalStaticSend = LocalStaticSend;
LocalStaticSend.attributeTypeMap = [
    {
        "name": "isLocal",
        "baseName": "is_local",
        "type": "LocalStaticSend.IsLocalEnum"
    },
    {
        "name": "sendPastRecipientsImmediately",
        "baseName": "send_past_recipients_immediately",
        "type": "boolean"
    }
];
(function (LocalStaticSend) {
    let IsLocalEnum;
    (function (IsLocalEnum) {
        IsLocalEnum[IsLocalEnum["True"] = true] = "True";
    })(IsLocalEnum = LocalStaticSend.IsLocalEnum || (LocalStaticSend.IsLocalEnum = {}));
})(LocalStaticSend = exports.LocalStaticSend || (exports.LocalStaticSend = {}));
