// klaviyo-api/dist/model/lowInventoryTrigger.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LowInventoryTrigger = void 0;
class LowInventoryTrigger {
    constructor() {
        this['timeframeDays'] = 30;
    }
    static getAttributeTypeMap() {
        return LowInventoryTrigger.attributeTypeMap;
    }
}
exports.LowInventoryTrigger = LowInventoryTrigger;
LowInventoryTrigger.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "LowInventoryEnum"
    },
    {
        "name": "productLevel",
        "baseName": "product_level",
        "type": "LowInventoryTrigger.ProductLevelEnum"
    },
    {
        "name": "triggerFilter",
        "baseName": "trigger_filter",
        "type": "LowInventoryConditionFilter"
    },
    {
        "name": "inventoryCount",
        "baseName": "inventory_count",
        "type": "number"
    },
    {
        "name": "audience",
        "baseName": "audience",
        "type": "Array<LowInventoryTrigger.AudienceEnum>"
    },
    {
        "name": "timeframeDays",
        "baseName": "timeframe_days",
        "type": "number"
    }
];
(function (LowInventoryTrigger) {
    let ProductLevelEnum;
    (function (ProductLevelEnum) {
        ProductLevelEnum[ProductLevelEnum["Product"] = 'product'] = "Product";
        ProductLevelEnum[ProductLevelEnum["Variant"] = 'variant'] = "Variant";
    })(ProductLevelEnum = LowInventoryTrigger.ProductLevelEnum || (LowInventoryTrigger.ProductLevelEnum = {}));
    let AudienceEnum;
    (function (AudienceEnum) {
        AudienceEnum[AudienceEnum["AddedToCart"] = 'added-to-cart'] = "AddedToCart";
        AudienceEnum[AudienceEnum["CheckoutStarted"] = 'checkout-started'] = "CheckoutStarted";
        AudienceEnum[AudienceEnum["Viewed"] = 'viewed'] = "Viewed";
    })(AudienceEnum = LowInventoryTrigger.AudienceEnum || (LowInventoryTrigger.AudienceEnum = {}));
})(LowInventoryTrigger = exports.LowInventoryTrigger || (exports.LowInventoryTrigger = {}));
