// klaviyo-api/dist/model/mobilePushMessageContent.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobilePushMessageContent = void 0;
class MobilePushMessageContent {
    constructor() {
        this['sound'] = false;
        this['badge'] = false;
        this['onOpen'] = 'home';
    }
    static getAttributeTypeMap() {
        return MobilePushMessageContent.attributeTypeMap;
    }
}
exports.MobilePushMessageContent = MobilePushMessageContent;
MobilePushMessageContent.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "body",
        "baseName": "body",
        "type": "string"
    },
    {
        "name": "sound",
        "baseName": "sound",
        "type": "boolean"
    },
    {
        "name": "badge",
        "baseName": "badge",
        "type": "boolean"
    },
    {
        "name": "mediaUrl",
        "baseName": "media_url",
        "type": "string"
    },
    {
        "name": "dynamicImage",
        "baseName": "dynamic_image",
        "type": "string"
    },
    {
        "name": "iosLink",
        "baseName": "ios_link",
        "type": "string"
    },
    {
        "name": "androidLink",
        "baseName": "android_link",
        "type": "string"
    },
    {
        "name": "onOpen",
        "baseName": "on_open",
        "type": "string"
    }
];
