// klaviyo-api/dist/model/numericOperatorFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NumericOperatorFilter = void 0;
class NumericOperatorFilter {
    static getAttributeTypeMap() {
        return NumericOperatorFilter.attributeTypeMap;
    }
}
exports.NumericOperatorFilter = NumericOperatorFilter;
NumericOperatorFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "NumericEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "NumericOperatorFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (NumericOperatorFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Equals"] = 'equals'] = "Equals";
        OperatorEnum[OperatorEnum["GreaterThan"] = 'greater-than'] = "GreaterThan";
        OperatorEnum[OperatorEnum["GreaterThanOrEqual"] = 'greater-than-or-equal'] = "GreaterThanOrEqual";
        OperatorEnum[OperatorEnum["LessThan"] = 'less-than'] = "LessThan";
        OperatorEnum[OperatorEnum["LessThanOrEqual"] = 'less-than-or-equal'] = "LessThanOrEqual";
        OperatorEnum[OperatorEnum["NotEquals"] = 'not-equals'] = "NotEquals";
    })(OperatorEnum = NumericOperatorFilter.OperatorEnum || (NumericOperatorFilter.OperatorEnum = {}));
})(NumericOperatorFilter = exports.NumericOperatorFilter || (exports.NumericOperatorFilter = {}));
