// klaviyo-api/dist/model/priceDropTrigger.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceDropTrigger = void 0;
class PriceDropTrigger {
    constructor() {
        /**
        * Price Drop amount type.
        */
        this['priceDropAmountUnit'] = PriceDropTrigger.PriceDropAmountUnitEnum.Currency;
        this['timeframeDays'] = 30;
        /**
        * Currency type.
        */
        this['currencyType'] = PriceDropTrigger.CurrencyTypeEnum.Usd;
    }
    static getAttributeTypeMap() {
        return PriceDropTrigger.attributeTypeMap;
    }
}
exports.PriceDropTrigger = PriceDropTrigger;
PriceDropTrigger.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "PriceDropEnum"
    },
    {
        "name": "triggerFilter",
        "baseName": "trigger_filter",
        "type": "PriceDropConditionFilter"
    },
    {
        "name": "priceDropAmountValue",
        "baseName": "price_drop_amount_value",
        "type": "number"
    },
    {
        "name": "priceDropAmountUnit",
        "baseName": "price_drop_amount_unit",
        "type": "PriceDropTrigger.PriceDropAmountUnitEnum"
    },
    {
        "name": "audience",
        "baseName": "audience",
        "type": "Array<PriceDropTrigger.AudienceEnum>"
    },
    {
        "name": "timeframeDays",
        "baseName": "timeframe_days",
        "type": "number"
    },
    {
        "name": "currencyType",
        "baseName": "currency_type",
        "type": "PriceDropTrigger.CurrencyTypeEnum"
    }
];
(function (PriceDropTrigger) {
    let PriceDropAmountUnitEnum;
    (function (PriceDropAmountUnitEnum) {
        PriceDropAmountUnitEnum[PriceDropAmountUnitEnum["Currency"] = 'currency'] = "Currency";
        PriceDropAmountUnitEnum[PriceDropAmountUnitEnum["Percent"] = 'percent'] = "Percent";
    })(PriceDropAmountUnitEnum = PriceDropTrigger.PriceDropAmountUnitEnum || (PriceDropTrigger.PriceDropAmountUnitEnum = {}));
    let AudienceEnum;
    (function (AudienceEnum) {
        AudienceEnum[AudienceEnum["CheckoutStarted"] = 'checkout-started'] = "CheckoutStarted";
        AudienceEnum[AudienceEnum["Viewed"] = 'viewed'] = "Viewed";
    })(AudienceEnum = PriceDropTrigger.AudienceEnum || (PriceDropTrigger.AudienceEnum = {}));
    let CurrencyTypeEnum;
    (function (CurrencyTypeEnum) {
        CurrencyTypeEnum[CurrencyTypeEnum["Usd"] = 'usd'] = "Usd";
    })(CurrencyTypeEnum = PriceDropTrigger.CurrencyTypeEnum || (PriceDropTrigger.CurrencyTypeEnum = {}));
})(PriceDropTrigger = exports.PriceDropTrigger || (exports.PriceDropTrigger = {}));
