// klaviyo-api/dist/model/profileImportJobResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileImportJobResponseObjectResourceAttributes = void 0;
class ProfileImportJobResponseObjectResourceAttributes {
    constructor() {
        /**
        * The total number of operations that have been completed by the job.
        */
        this['completedCount'] = 0;
        /**
        * The total number of operations that have failed as part of the job.
        */
        this['failedCount'] = 0;
    }
    static getAttributeTypeMap() {
        return ProfileImportJobResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.ProfileImportJobResponseObjectResourceAttributes = ProfileImportJobResponseObjectResourceAttributes;
ProfileImportJobResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "ProfileImportJobResponseObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "createdAt",
        "baseName": "created_at",
        "type": "Date"
    },
    {
        "name": "totalCount",
        "baseName": "total_count",
        "type": "number"
    },
    {
        "name": "completedCount",
        "baseName": "completed_count",
        "type": "number"
    },
    {
        "name": "failedCount",
        "baseName": "failed_count",
        "type": "number"
    },
    {
        "name": "completedAt",
        "baseName": "completed_at",
        "type": "Date"
    },
    {
        "name": "expiresAt",
        "baseName": "expires_at",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "started_at",
        "type": "Date"
    }
];
(function (ProfileImportJobResponseObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Cancelled"] = 'cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["Complete"] = 'complete'] = "Complete";
        StatusEnum[StatusEnum["Processing"] = 'processing'] = "Processing";
        StatusEnum[StatusEnum["Queued"] = 'queued'] = "Queued";
    })(StatusEnum = ProfileImportJobResponseObjectResourceAttributes.StatusEnum || (ProfileImportJobResponseObjectResourceAttributes.StatusEnum = {}));
})(ProfileImportJobResponseObjectResourceAttributes = exports.ProfileImportJobResponseObjectResourceAttributes || (exports.ProfileImportJobResponseObjectResourceAttributes = {}));
