// klaviyo-api/dist/model/profileOperationUpdateOrCreateString.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileOperationUpdateOrCreateString = void 0;
class ProfileOperationUpdateOrCreateString {
    static getAttributeTypeMap() {
        return ProfileOperationUpdateOrCreateString.attributeTypeMap;
    }
}
exports.ProfileOperationUpdateOrCreateString = ProfileOperationUpdateOrCreateString;
ProfileOperationUpdateOrCreateString.attributeTypeMap = [
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfileOperationUpdateOrCreateString.OperatorEnum"
    },
    {
        "name": "propertyType",
        "baseName": "property_type",
        "type": "StringEnum"
    },
    {
        "name": "propertyKey",
        "baseName": "property_key",
        "type": "string"
    },
    {
        "name": "propertyValue",
        "baseName": "property_value",
        "type": "string"
    }
];
(function (ProfileOperationUpdateOrCreateString) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Create"] = 'create'] = "Create";
        OperatorEnum[OperatorEnum["Update"] = 'update'] = "Update";
    })(OperatorEnum = ProfileOperationUpdateOrCreateString.OperatorEnum || (ProfileOperationUpdateOrCreateString.OperatorEnum = {}));
})(ProfileOperationUpdateOrCreateString = exports.ProfileOperationUpdateOrCreateString || (exports.ProfileOperationUpdateOrCreateString = {}));
