// klaviyo-api/dist/model/profilePredictiveAnalyticsNumericCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePredictiveAnalyticsNumericCondition = void 0;
class ProfilePredictiveAnalyticsNumericCondition {
    static getAttributeTypeMap() {
        return ProfilePredictiveAnalyticsNumericCondition.attributeTypeMap;
    }
}
exports.ProfilePredictiveAnalyticsNumericCondition = ProfilePredictiveAnalyticsNumericCondition;
ProfilePredictiveAnalyticsNumericCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfilePredictiveAnalyticsEnum"
    },
    {
        "name": "dimension",
        "baseName": "dimension",
        "type": "ProfilePredictiveAnalyticsNumericCondition.DimensionEnum"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "NumericOperatorFilter"
    }
];
(function (ProfilePredictiveAnalyticsNumericCondition) {
    let DimensionEnum;
    (function (DimensionEnum) {
        DimensionEnum[DimensionEnum["AverageDaysBetweenOrders"] = 'average_days_between_orders'] = "AverageDaysBetweenOrders";
        DimensionEnum[DimensionEnum["AverageOrderValue"] = 'average_order_value'] = "AverageOrderValue";
        DimensionEnum[DimensionEnum["ChurnProbability"] = 'churn_probability'] = "ChurnProbability";
        DimensionEnum[DimensionEnum["HistoricClv"] = 'historic_clv'] = "HistoricClv";
        DimensionEnum[DimensionEnum["HistoricNumberOfOrders"] = 'historic_number_of_orders'] = "HistoricNumberOfOrders";
        DimensionEnum[DimensionEnum["PredictedClv"] = 'predicted_clv'] = "PredictedClv";
        DimensionEnum[DimensionEnum["PredictedNumberOfOrders"] = 'predicted_number_of_orders'] = "PredictedNumberOfOrders";
        DimensionEnum[DimensionEnum["TotalClv"] = 'total_clv'] = "TotalClv";
    })(DimensionEnum = ProfilePredictiveAnalyticsNumericCondition.DimensionEnum || (ProfilePredictiveAnalyticsNumericCondition.DimensionEnum = {}));
})(ProfilePredictiveAnalyticsNumericCondition = exports.ProfilePredictiveAnalyticsNumericCondition || (exports.ProfilePredictiveAnalyticsNumericCondition = {}));
