// klaviyo-api/dist/model/profilePostalCodeDistanceCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePostalCodeDistanceCondition = void 0;
class ProfilePostalCodeDistanceCondition {
    static getAttributeTypeMap() {
        return ProfilePostalCodeDistanceCondition.attributeTypeMap;
    }
}
exports.ProfilePostalCodeDistanceCondition = ProfilePostalCodeDistanceCondition;
ProfilePostalCodeDistanceCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfilePostalCodeDistanceEnum"
    },
    {
        "name": "countryCode",
        "baseName": "country_code",
        "type": "string"
    },
    {
        "name": "postalCode",
        "baseName": "postal_code",
        "type": "string"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "ProfilePostalCodeDistanceCondition.UnitEnum"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "ProfilePostalCodeDistanceConditionFilter"
    }
];
(function (ProfilePostalCodeDistanceCondition) {
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Kilometers"] = 'kilometers'] = "Kilometers";
        UnitEnum[UnitEnum["Miles"] = 'miles'] = "Miles";
    })(UnitEnum = ProfilePostalCodeDistanceCondition.UnitEnum || (ProfilePostalCodeDistanceCondition.UnitEnum = {}));
})(ProfilePostalCodeDistanceCondition = exports.ProfilePostalCodeDistanceCondition || (exports.ProfilePostalCodeDistanceCondition = {}));
