// klaviyo-api/dist/model/profileOperationUpdateOrCreateList.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileOperationUpdateOrCreateList = void 0;
class ProfileOperationUpdateOrCreateList {
    static getAttributeTypeMap() {
        return ProfileOperationUpdateOrCreateList.attributeTypeMap;
    }
}
exports.ProfileOperationUpdateOrCreateList = ProfileOperationUpdateOrCreateList;
ProfileOperationUpdateOrCreateList.attributeTypeMap = [
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfileOperationUpdateOrCreateList.OperatorEnum"
    },
    {
        "name": "propertyType",
        "baseName": "property_type",
        "type": "ListEnum"
    },
    {
        "name": "propertyOperation",
        "baseName": "property_operation",
        "type": "ProfileOperationUpdateOrCreateList.PropertyOperationEnum"
    },
    {
        "name": "propertyKey",
        "baseName": "property_key",
        "type": "string"
    },
    {
        "name": "propertyValue",
        "baseName": "property_value",
        "type": "string"
    }
];
(function (ProfileOperationUpdateOrCreateList) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Create"] = 'create'] = "Create";
        OperatorEnum[OperatorEnum["Update"] = 'update'] = "Update";
    })(OperatorEnum = ProfileOperationUpdateOrCreateList.OperatorEnum || (ProfileOperationUpdateOrCreateList.OperatorEnum = {}));
    let PropertyOperationEnum;
    (function (PropertyOperationEnum) {
        PropertyOperationEnum[PropertyOperationEnum["Add"] = 'add'] = "Add";
        PropertyOperationEnum[PropertyOperationEnum["Remove"] = 'remove'] = "Remove";
    })(PropertyOperationEnum = ProfileOperationUpdateOrCreateList.PropertyOperationEnum || (ProfileOperationUpdateOrCreateList.PropertyOperationEnum = {}));
})(ProfileOperationUpdateOrCreateList = exports.ProfileOperationUpdateOrCreateList || (exports.ProfileOperationUpdateOrCreateList = {}));
