// klaviyo-api/dist/model/profileOperationUpdateOrCreateBoolean.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileOperationUpdateOrCreateBoolean = void 0;
class ProfileOperationUpdateOrCreateBoolean {
    static getAttributeTypeMap() {
        return ProfileOperationUpdateOrCreateBoolean.attributeTypeMap;
    }
}
exports.ProfileOperationUpdateOrCreateBoolean = ProfileOperationUpdateOrCreateBoolean;
ProfileOperationUpdateOrCreateBoolean.attributeTypeMap = [
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfileOperationUpdateOrCreateBoolean.OperatorEnum"
    },
    {
        "name": "propertyType",
        "baseName": "property_type",
        "type": "BooleanEnum"
    },
    {
        "name": "propertyKey",
        "baseName": "property_key",
        "type": "string"
    },
    {
        "name": "propertyValue",
        "baseName": "property_value",
        "type": "boolean"
    }
];
(function (ProfileOperationUpdateOrCreateBoolean) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Create"] = 'create'] = "Create";
        OperatorEnum[OperatorEnum["Update"] = 'update'] = "Update";
    })(OperatorEnum = ProfileOperationUpdateOrCreateBoolean.OperatorEnum || (ProfileOperationUpdateOrCreateBoolean.OperatorEnum = {}));
})(ProfileOperationUpdateOrCreateBoolean = exports.ProfileOperationUpdateOrCreateBoolean || (exports.ProfileOperationUpdateOrCreateBoolean = {}));
