// klaviyo-api/dist/model/profileOperationUpdateOrCreateNumeric.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileOperationUpdateOrCreateNumeric = void 0;
class ProfileOperationUpdateOrCreateNumeric {
    static getAttributeTypeMap() {
        return ProfileOperationUpdateOrCreateNumeric.attributeTypeMap;
    }
}
exports.ProfileOperationUpdateOrCreateNumeric = ProfileOperationUpdateOrCreateNumeric;
ProfileOperationUpdateOrCreateNumeric.attributeTypeMap = [
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfileOperationUpdateOrCreateNumeric.OperatorEnum"
    },
    {
        "name": "propertyType",
        "baseName": "property_type",
        "type": "NumericEnum"
    },
    {
        "name": "propertyKey",
        "baseName": "property_key",
        "type": "string"
    },
    {
        "name": "propertyValue",
        "baseName": "property_value",
        "type": "number"
    }
];
(function (ProfileOperationUpdateOrCreateNumeric) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Create"] = 'create'] = "Create";
        OperatorEnum[OperatorEnum["Update"] = 'update'] = "Update";
    })(OperatorEnum = ProfileOperationUpdateOrCreateNumeric.OperatorEnum || (ProfileOperationUpdateOrCreateNumeric.OperatorEnum = {}));
})(ProfileOperationUpdateOrCreateNumeric = exports.ProfileOperationUpdateOrCreateNumeric || (exports.ProfileOperationUpdateOrCreateNumeric = {}));
