// klaviyo-api/dist/model/profileRegionCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileRegionCondition = void 0;
class ProfileRegionCondition {
    static getAttributeTypeMap() {
        return ProfileRegionCondition.attributeTypeMap;
    }
}
exports.ProfileRegionCondition = ProfileRegionCondition;
ProfileRegionCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfileRegionEnum"
    },
    {
        "name": "inRegion",
        "baseName": "in_region",
        "type": "boolean"
    },
    {
        "name": "region",
        "baseName": "region",
        "type": "ProfileRegionCondition.RegionEnum"
    }
];
(function (ProfileRegionCondition) {
    let RegionEnum;
    (function (RegionEnum) {
        RegionEnum[RegionEnum["EuropeanUnion"] = 'european_union'] = "EuropeanUnion";
        RegionEnum[RegionEnum["UnitedStates"] = 'united_states'] = "UnitedStates";
    })(RegionEnum = ProfileRegionCondition.RegionEnum || (ProfileRegionCondition.RegionEnum = {}));
})(ProfileRegionCondition = exports.ProfileRegionCondition || (exports.ProfileRegionCondition = {}));
