// klaviyo-api/dist/model/profilePredictiveAnalyticsStringFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePredictiveAnalyticsStringFilter = void 0;
class ProfilePredictiveAnalyticsStringFilter {
    static getAttributeTypeMap() {
        return ProfilePredictiveAnalyticsStringFilter.attributeTypeMap;
    }
}
exports.ProfilePredictiveAnalyticsStringFilter = ProfilePredictiveAnalyticsStringFilter;
ProfilePredictiveAnalyticsStringFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "StringEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfilePredictiveAnalyticsStringFilterOperator"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "ProfilePredictiveAnalyticsStringFilter.ValueEnum"
    }
];
(function (ProfilePredictiveAnalyticsStringFilter) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum[ValueEnum["LikelyFemale"] = 'likely_female'] = "LikelyFemale";
        ValueEnum[ValueEnum["LikelyMale"] = 'likely_male'] = "LikelyMale";
        ValueEnum[ValueEnum["Uncertain"] = 'uncertain'] = "Uncertain";
    })(ValueEnum = ProfilePredictiveAnalyticsStringFilter.ValueEnum || (ProfilePredictiveAnalyticsStringFilter.ValueEnum = {}));
})(ProfilePredictiveAnalyticsStringFilter = exports.ProfilePredictiveAnalyticsStringFilter || (exports.ProfilePredictiveAnalyticsStringFilter = {}));
