// klaviyo-api/dist/model/profileOperationUpdateOrCreateDate.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileOperationUpdateOrCreateDate = void 0;
class ProfileOperationUpdateOrCreateDate {
    static getAttributeTypeMap() {
        return ProfileOperationUpdateOrCreateDate.attributeTypeMap;
    }
}
exports.ProfileOperationUpdateOrCreateDate = ProfileOperationUpdateOrCreateDate;
ProfileOperationUpdateOrCreateDate.attributeTypeMap = [
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfileOperationUpdateOrCreateDate.OperatorEnum"
    },
    {
        "name": "propertyType",
        "baseName": "property_type",
        "type": "DateEnum"
    },
    {
        "name": "propertyKey",
        "baseName": "property_key",
        "type": "string"
    },
    {
        "name": "propertyValue",
        "baseName": "property_value",
        "type": "string"
    }
];
(function (ProfileOperationUpdateOrCreateDate) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Create"] = 'create'] = "Create";
        OperatorEnum[OperatorEnum["Update"] = 'update'] = "Update";
    })(OperatorEnum = ProfileOperationUpdateOrCreateDate.OperatorEnum || (ProfileOperationUpdateOrCreateDate.OperatorEnum = {}));
})(ProfileOperationUpdateOrCreateDate = exports.ProfileOperationUpdateOrCreateDate || (exports.ProfileOperationUpdateOrCreateDate = {}));
