// klaviyo-api/dist/model/pushTokenCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushTokenCreateQueryResourceObjectAttributes = void 0;
class PushTokenCreateQueryResourceObjectAttributes {
    constructor() {
        /**
        * This is the enablement status for the individual push token.
        */
        this['enablementStatus'] = PushTokenCreateQueryResourceObjectAttributes.EnablementStatusEnum.Authorized;
        /**
        * The background state of the push token.
        */
        this['background'] = PushTokenCreateQueryResourceObjectAttributes.BackgroundEnum.Available;
    }
    static getAttributeTypeMap() {
        return PushTokenCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.PushTokenCreateQueryResourceObjectAttributes = PushTokenCreateQueryResourceObjectAttributes;
PushTokenCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "token",
        "baseName": "token",
        "type": "string"
    },
    {
        "name": "platform",
        "baseName": "platform",
        "type": "PushTokenCreateQueryResourceObjectAttributes.PlatformEnum"
    },
    {
        "name": "enablementStatus",
        "baseName": "enablement_status",
        "type": "PushTokenCreateQueryResourceObjectAttributes.EnablementStatusEnum"
    },
    {
        "name": "vendor",
        "baseName": "vendor",
        "type": "PushTokenCreateQueryResourceObjectAttributes.VendorEnum"
    },
    {
        "name": "background",
        "baseName": "background",
        "type": "PushTokenCreateQueryResourceObjectAttributes.BackgroundEnum"
    },
    {
        "name": "deviceMetadata",
        "baseName": "device_metadata",
        "type": "DeviceMetadata"
    },
    {
        "name": "profile",
        "baseName": "profile",
        "type": "PushTokenCreateQueryResourceObjectAttributesProfile"
    }
];
(function (PushTokenCreateQueryResourceObjectAttributes) {
    let PlatformEnum;
    (function (PlatformEnum) {
        PlatformEnum[PlatformEnum["Android"] = 'android'] = "Android";
        PlatformEnum[PlatformEnum["Ios"] = 'ios'] = "Ios";
    })(PlatformEnum = PushTokenCreateQueryResourceObjectAttributes.PlatformEnum || (PushTokenCreateQueryResourceObjectAttributes.PlatformEnum = {}));
    let EnablementStatusEnum;
    (function (EnablementStatusEnum) {
        EnablementStatusEnum[EnablementStatusEnum["Authorized"] = 'AUTHORIZED'] = "Authorized";
        EnablementStatusEnum[EnablementStatusEnum["Denied"] = 'DENIED'] = "Denied";
        EnablementStatusEnum[EnablementStatusEnum["NotDetermined"] = 'NOT_DETERMINED'] = "NotDetermined";
        EnablementStatusEnum[EnablementStatusEnum["Provisional"] = 'PROVISIONAL'] = "Provisional";
        EnablementStatusEnum[EnablementStatusEnum["Unauthorized"] = 'UNAUTHORIZED'] = "Unauthorized";
    })(EnablementStatusEnum = PushTokenCreateQueryResourceObjectAttributes.EnablementStatusEnum || (PushTokenCreateQueryResourceObjectAttributes.EnablementStatusEnum = {}));
    let VendorEnum;
    (function (VendorEnum) {
        VendorEnum[VendorEnum["Apns"] = 'apns'] = "Apns";
        VendorEnum[VendorEnum["Fcm"] = 'fcm'] = "Fcm";
    })(VendorEnum = PushTokenCreateQueryResourceObjectAttributes.VendorEnum || (PushTokenCreateQueryResourceObjectAttributes.VendorEnum = {}));
    let BackgroundEnum;
    (function (BackgroundEnum) {
        BackgroundEnum[BackgroundEnum["Available"] = 'AVAILABLE'] = "Available";
        BackgroundEnum[BackgroundEnum["Denied"] = 'DENIED'] = "Denied";
        BackgroundEnum[BackgroundEnum["Restricted"] = 'RESTRICTED'] = "Restricted";
    })(BackgroundEnum = PushTokenCreateQueryResourceObjectAttributes.BackgroundEnum || (PushTokenCreateQueryResourceObjectAttributes.BackgroundEnum = {}));
})(PushTokenCreateQueryResourceObjectAttributes = exports.PushTokenCreateQueryResourceObjectAttributes || (exports.PushTokenCreateQueryResourceObjectAttributes = {}));
