// klaviyo-api/dist/model/pushTokenResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushTokenResponseObjectResourceAttributes = void 0;
class PushTokenResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return PushTokenResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.PushTokenResponseObjectResourceAttributes = PushTokenResponseObjectResourceAttributes;
PushTokenResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "token",
        "baseName": "token",
        "type": "string"
    },
    {
        "name": "enablementStatus",
        "baseName": "enablement_status",
        "type": "PushTokenResponseObjectResourceAttributes.EnablementStatusEnum"
    },
    {
        "name": "platform",
        "baseName": "platform",
        "type": "PushTokenResponseObjectResourceAttributes.PlatformEnum"
    },
    {
        "name": "vendor",
        "baseName": "vendor",
        "type": "string"
    },
    {
        "name": "background",
        "baseName": "background",
        "type": "string"
    },
    {
        "name": "recordedDate",
        "baseName": "recorded_date",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "DeviceMetadata"
    }
];
(function (PushTokenResponseObjectResourceAttributes) {
    let EnablementStatusEnum;
    (function (EnablementStatusEnum) {
        EnablementStatusEnum[EnablementStatusEnum["Authorized"] = 'AUTHORIZED'] = "Authorized";
        EnablementStatusEnum[EnablementStatusEnum["Denied"] = 'DENIED'] = "Denied";
        EnablementStatusEnum[EnablementStatusEnum["NotDetermined"] = 'NOT_DETERMINED'] = "NotDetermined";
        EnablementStatusEnum[EnablementStatusEnum["Provisional"] = 'PROVISIONAL'] = "Provisional";
        EnablementStatusEnum[EnablementStatusEnum["Unauthorized"] = 'UNAUTHORIZED'] = "Unauthorized";
    })(EnablementStatusEnum = PushTokenResponseObjectResourceAttributes.EnablementStatusEnum || (PushTokenResponseObjectResourceAttributes.EnablementStatusEnum = {}));
    let PlatformEnum;
    (function (PlatformEnum) {
        PlatformEnum[PlatformEnum["Android"] = 'android'] = "Android";
        PlatformEnum[PlatformEnum["Ios"] = 'ios'] = "Ios";
    })(PlatformEnum = PushTokenResponseObjectResourceAttributes.PlatformEnum || (PushTokenResponseObjectResourceAttributes.PlatformEnum = {}));
})(PushTokenResponseObjectResourceAttributes = exports.PushTokenResponseObjectResourceAttributes || (exports.PushTokenResponseObjectResourceAttributes = {}));
