// klaviyo-api/dist/model/relativeDateRangeFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeDateRangeFilter = void 0;
class RelativeDateRangeFilter {
    static getAttributeTypeMap() {
        return RelativeDateRangeFilter.attributeTypeMap;
    }
}
exports.RelativeDateRangeFilter = RelativeDateRangeFilter;
RelativeDateRangeFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "RelativeDateRangeFilter.OperatorEnum"
    },
    {
        "name": "start",
        "baseName": "start",
        "type": "number"
    },
    {
        "name": "end",
        "baseName": "end",
        "type": "number"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "RelativeDateRangeFilter.UnitEnum"
    }
];
(function (RelativeDateRangeFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Between"] = 'between'] = "Between";
    })(OperatorEnum = RelativeDateRangeFilter.OperatorEnum || (RelativeDateRangeFilter.OperatorEnum = {}));
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Day"] = 'day'] = "Day";
        UnitEnum[UnitEnum["Hour"] = 'hour'] = "Hour";
        UnitEnum[UnitEnum["Week"] = 'week'] = "Week";
    })(UnitEnum = RelativeDateRangeFilter.UnitEnum || (RelativeDateRangeFilter.UnitEnum = {}));
})(RelativeDateRangeFilter = exports.RelativeDateRangeFilter || (exports.RelativeDateRangeFilter = {}));
