// klaviyo-api/dist/model/relativeDateOperatorBaseFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeDateOperatorBaseFilter = void 0;
class RelativeDateOperatorBaseFilter {
    static getAttributeTypeMap() {
        return RelativeDateOperatorBaseFilter.attributeTypeMap;
    }
}
exports.RelativeDateOperatorBaseFilter = RelativeDateOperatorBaseFilter;
RelativeDateOperatorBaseFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "RelativeDateOperatorBaseFilter.OperatorEnum"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "RelativeDateOperatorBaseFilter.UnitEnum"
    },
    {
        "name": "quantity",
        "baseName": "quantity",
        "type": "number"
    }
];
(function (RelativeDateOperatorBaseFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["AtLeast"] = 'at-least'] = "AtLeast";
        OperatorEnum[OperatorEnum["InTheLast"] = 'in-the-last'] = "InTheLast";
        OperatorEnum[OperatorEnum["InTheNext"] = 'in-the-next'] = "InTheNext";
    })(OperatorEnum = RelativeDateOperatorBaseFilter.OperatorEnum || (RelativeDateOperatorBaseFilter.OperatorEnum = {}));
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Day"] = 'day'] = "Day";
        UnitEnum[UnitEnum["Hour"] = 'hour'] = "Hour";
        UnitEnum[UnitEnum["Week"] = 'week'] = "Week";
    })(UnitEnum = RelativeDateOperatorBaseFilter.UnitEnum || (RelativeDateOperatorBaseFilter.UnitEnum = {}));
})(RelativeDateOperatorBaseFilter = exports.RelativeDateOperatorBaseFilter || (exports.RelativeDateOperatorBaseFilter = {}));
