// klaviyo-api/dist/model/relativeAnniversaryDateFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeAnniversaryDateFilter = void 0;
class RelativeAnniversaryDateFilter {
    static getAttributeTypeMap() {
        return RelativeAnniversaryDateFilter.attributeTypeMap;
    }
}
exports.RelativeAnniversaryDateFilter = RelativeAnniversaryDateFilter;
RelativeAnniversaryDateFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "RelativeAnniversaryDateFilter.OperatorEnum"
    },
    {
        "name": "unit",
        "baseName": "unit",
        "type": "RelativeAnniversaryDateFilter.UnitEnum"
    },
    {
        "name": "quantity",
        "baseName": "quantity",
        "type": "number"
    }
];
(function (RelativeAnniversaryDateFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Last"] = 'anniversary-last'] = "Last";
        OperatorEnum[OperatorEnum["Next"] = 'anniversary-next'] = "Next";
    })(OperatorEnum = RelativeAnniversaryDateFilter.OperatorEnum || (RelativeAnniversaryDateFilter.OperatorEnum = {}));
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum[UnitEnum["Day"] = 'day'] = "Day";
        UnitEnum[UnitEnum["Hour"] = 'hour'] = "Hour";
        UnitEnum[UnitEnum["Week"] = 'week'] = "Week";
    })(UnitEnum = RelativeAnniversaryDateFilter.UnitEnum || (RelativeAnniversaryDateFilter.UnitEnum = {}));
})(RelativeAnniversaryDateFilter = exports.RelativeAnniversaryDateFilter || (exports.RelativeAnniversaryDateFilter = {}));
