// klaviyo-api/dist/model/reviewResponseDTOObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReviewResponseDTOObjectResourceAttributes = void 0;
class ReviewResponseDTOObjectResourceAttributes {
    static getAttributeTypeMap() {
        return ReviewResponseDTOObjectResourceAttributes.attributeTypeMap;
    }
}
exports.ReviewResponseDTOObjectResourceAttributes = ReviewResponseDTOObjectResourceAttributes;
ReviewResponseDTOObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "ReviewResponseDTOObjectResourceAttributesStatus"
    },
    {
        "name": "verified",
        "baseName": "verified",
        "type": "boolean"
    },
    {
        "name": "reviewType",
        "baseName": "review_type",
        "type": "ReviewResponseDTOObjectResourceAttributes.ReviewTypeEnum"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "images",
        "baseName": "images",
        "type": "Array<string>"
    },
    {
        "name": "product",
        "baseName": "product",
        "type": "ReviewProductDTO"
    },
    {
        "name": "rating",
        "baseName": "rating",
        "type": "number"
    },
    {
        "name": "author",
        "baseName": "author",
        "type": "string"
    },
    {
        "name": "content",
        "baseName": "content",
        "type": "string"
    },
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "smartQuote",
        "baseName": "smart_quote",
        "type": "string"
    },
    {
        "name": "publicReply",
        "baseName": "public_reply",
        "type": "ReviewPublicReply"
    }
];
(function (ReviewResponseDTOObjectResourceAttributes) {
    let ReviewTypeEnum;
    (function (ReviewTypeEnum) {
        ReviewTypeEnum[ReviewTypeEnum["Question"] = 'question'] = "Question";
        ReviewTypeEnum[ReviewTypeEnum["Rating"] = 'rating'] = "Rating";
        ReviewTypeEnum[ReviewTypeEnum["Review"] = 'review'] = "Review";
        ReviewTypeEnum[ReviewTypeEnum["Store"] = 'store'] = "Store";
    })(ReviewTypeEnum = ReviewResponseDTOObjectResourceAttributes.ReviewTypeEnum || (ReviewResponseDTOObjectResourceAttributes.ReviewTypeEnum = {}));
})(ReviewResponseDTOObjectResourceAttributes = exports.ReviewResponseDTOObjectResourceAttributes || (exports.ReviewResponseDTOObjectResourceAttributes = {}));
