// klaviyo-api/dist/model/segmentSeriesRequestDTOResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SegmentSeriesRequestDTOResourceObjectAttributes = void 0;
class SegmentSeriesRequestDTOResourceObjectAttributes {
    static getAttributeTypeMap() {
        return SegmentSeriesRequestDTOResourceObjectAttributes.attributeTypeMap;
    }
}
exports.SegmentSeriesRequestDTOResourceObjectAttributes = SegmentSeriesRequestDTOResourceObjectAttributes;
SegmentSeriesRequestDTOResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "statistics",
        "baseName": "statistics",
        "type": "Array<SegmentSeriesRequestDTOResourceObjectAttributes.StatisticsEnum>"
    },
    {
        "name": "timeframe",
        "baseName": "timeframe",
        "type": "SegmentValuesRequestDTOResourceObjectAttributesTimeframe"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "SegmentSeriesRequestDTOResourceObjectAttributes.IntervalEnum"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "string"
    }
];
(function (SegmentSeriesRequestDTOResourceObjectAttributes) {
    let StatisticsEnum;
    (function (StatisticsEnum) {
        StatisticsEnum[StatisticsEnum["MembersAdded"] = 'members_added'] = "MembersAdded";
        StatisticsEnum[StatisticsEnum["MembersRemoved"] = 'members_removed'] = "MembersRemoved";
        StatisticsEnum[StatisticsEnum["NetMembersChanged"] = 'net_members_changed'] = "NetMembersChanged";
        StatisticsEnum[StatisticsEnum["TotalMembers"] = 'total_members'] = "TotalMembers";
    })(StatisticsEnum = SegmentSeriesRequestDTOResourceObjectAttributes.StatisticsEnum || (SegmentSeriesRequestDTOResourceObjectAttributes.StatisticsEnum = {}));
    let IntervalEnum;
    (function (IntervalEnum) {
        IntervalEnum[IntervalEnum["Daily"] = 'daily'] = "Daily";
        IntervalEnum[IntervalEnum["Hourly"] = 'hourly'] = "Hourly";
        IntervalEnum[IntervalEnum["Monthly"] = 'monthly'] = "Monthly";
        IntervalEnum[IntervalEnum["Weekly"] = 'weekly'] = "Weekly";
    })(IntervalEnum = SegmentSeriesRequestDTOResourceObjectAttributes.IntervalEnum || (SegmentSeriesRequestDTOResourceObjectAttributes.IntervalEnum = {}));
})(SegmentSeriesRequestDTOResourceObjectAttributes = exports.SegmentSeriesRequestDTOResourceObjectAttributes || (exports.SegmentSeriesRequestDTOResourceObjectAttributes = {}));
