// klaviyo-api/dist/model/segmentsProfileMetricCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SegmentsProfileMetricCondition = void 0;
class SegmentsProfileMetricCondition {
    static getAttributeTypeMap() {
        return SegmentsProfileMetricCondition.attributeTypeMap;
    }
}
exports.SegmentsProfileMetricCondition = SegmentsProfileMetricCondition;
SegmentsProfileMetricCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfileMetricEnum"
    },
    {
        "name": "metricId",
        "baseName": "metric_id",
        "type": "string"
    },
    {
        "name": "measurement",
        "baseName": "measurement",
        "type": "SegmentsProfileMetricCondition.MeasurementEnum"
    },
    {
        "name": "measurementFilter",
        "baseName": "measurement_filter",
        "type": "NumericOperatorFilter"
    },
    {
        "name": "timeframeFilter",
        "baseName": "timeframe_filter",
        "type": "SegmentsProfileMetricConditionTimeframeFilter"
    },
    {
        "name": "metricFilters",
        "baseName": "metric_filters",
        "type": "Array<ProfileMetricPropertyFilter>"
    }
];
(function (SegmentsProfileMetricCondition) {
    let MeasurementEnum;
    (function (MeasurementEnum) {
        MeasurementEnum[MeasurementEnum["Count"] = 'count'] = "Count";
        MeasurementEnum[MeasurementEnum["Sum"] = 'sum'] = "Sum";
    })(MeasurementEnum = SegmentsProfileMetricCondition.MeasurementEnum || (SegmentsProfileMetricCondition.MeasurementEnum = {}));
})(SegmentsProfileMetricCondition = exports.SegmentsProfileMetricCondition || (exports.SegmentsProfileMetricCondition = {}));
