// klaviyo-api/dist/model/segmentValuesRequestDTOResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SegmentValuesRequestDTOResourceObjectAttributes = void 0;
class SegmentValuesRequestDTOResourceObjectAttributes {
    static getAttributeTypeMap() {
        return SegmentValuesRequestDTOResourceObjectAttributes.attributeTypeMap;
    }
}
exports.SegmentValuesRequestDTOResourceObjectAttributes = SegmentValuesRequestDTOResourceObjectAttributes;
SegmentValuesRequestDTOResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "statistics",
        "baseName": "statistics",
        "type": "Array<SegmentValuesRequestDTOResourceObjectAttributes.StatisticsEnum>"
    },
    {
        "name": "timeframe",
        "baseName": "timeframe",
        "type": "SegmentValuesRequestDTOResourceObjectAttributesTimeframe"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "string"
    }
];
(function (SegmentValuesRequestDTOResourceObjectAttributes) {
    let StatisticsEnum;
    (function (StatisticsEnum) {
        StatisticsEnum[StatisticsEnum["MembersAdded"] = 'members_added'] = "MembersAdded";
        StatisticsEnum[StatisticsEnum["MembersRemoved"] = 'members_removed'] = "MembersRemoved";
        StatisticsEnum[StatisticsEnum["NetMembersChanged"] = 'net_members_changed'] = "NetMembersChanged";
        StatisticsEnum[StatisticsEnum["TotalMembers"] = 'total_members'] = "TotalMembers";
    })(StatisticsEnum = SegmentValuesRequestDTOResourceObjectAttributes.StatisticsEnum || (SegmentValuesRequestDTOResourceObjectAttributes.StatisticsEnum = {}));
})(SegmentValuesRequestDTOResourceObjectAttributes = exports.SegmentValuesRequestDTOResourceObjectAttributes || (exports.SegmentValuesRequestDTOResourceObjectAttributes = {}));
