// klaviyo-api/dist/model/sendWebhookActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendWebhookActionData = void 0;
class SendWebhookActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendWebhookActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendWebhookActionData.attributeTypeMap;
    }
}
exports.SendWebhookActionData = SendWebhookActionData;
SendWebhookActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowWebhook"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendWebhookActionData.StatusEnum"
    }
];
(function (SendWebhookActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendWebhookActionData.StatusEnum || (SendWebhookActionData.StatusEnum = {}));
})(SendWebhookActionData = exports.SendWebhookActionData || (exports.SendWebhookActionData = {}));
