// klaviyo-api/dist/model/sendPushNotificationActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendPushNotificationActionData = void 0;
class SendPushNotificationActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendPushNotificationActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendPushNotificationActionData.attributeTypeMap;
    }
}
exports.SendPushNotificationActionData = SendPushNotificationActionData;
SendPushNotificationActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowPushNotification"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendPushNotificationActionData.StatusEnum"
    }
];
(function (SendPushNotificationActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendPushNotificationActionData.StatusEnum || (SendPushNotificationActionData.StatusEnum = {}));
})(SendPushNotificationActionData = exports.SendPushNotificationActionData || (exports.SendPushNotificationActionData = {}));
