// klaviyo-api/dist/model/sendInternalAlertActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendInternalAlertActionData = void 0;
class SendInternalAlertActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendInternalAlertActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendInternalAlertActionData.attributeTypeMap;
    }
}
exports.SendInternalAlertActionData = SendInternalAlertActionData;
SendInternalAlertActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowInternalAlert"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendInternalAlertActionData.StatusEnum"
    }
];
(function (SendInternalAlertActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendInternalAlertActionData.StatusEnum || (SendInternalAlertActionData.StatusEnum = {}));
})(SendInternalAlertActionData = exports.SendInternalAlertActionData || (exports.SendInternalAlertActionData = {}));
