// klaviyo-api/dist/model/sendEmailActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendEmailActionData = void 0;
class SendEmailActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendEmailActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendEmailActionData.attributeTypeMap;
    }
}
exports.SendEmailActionData = SendEmailActionData;
SendEmailActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowEmail"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendEmailActionData.StatusEnum"
    }
];
(function (SendEmailActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendEmailActionData.StatusEnum || (SendEmailActionData.StatusEnum = {}));
})(SendEmailActionData = exports.SendEmailActionData || (exports.SendEmailActionData = {}));
