// klaviyo-api/dist/model/sendSmsActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendSmsActionData = void 0;
class SendSmsActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendSmsActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendSmsActionData.attributeTypeMap;
    }
}
exports.SendSmsActionData = SendSmsActionData;
SendSmsActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowSms"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendSmsActionData.StatusEnum"
    }
];
(function (SendSmsActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendSmsActionData.StatusEnum || (SendSmsActionData.StatusEnum = {}));
})(SendSmsActionData = exports.SendSmsActionData || (exports.SendSmsActionData = {}));
