// klaviyo-api/dist/model/serverBISSubscriptionCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerBISSubscriptionCreateQueryResourceObjectAttributes = void 0;
class ServerBISSubscriptionCreateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return ServerBISSubscriptionCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.ServerBISSubscriptionCreateQueryResourceObjectAttributes = ServerBISSubscriptionCreateQueryResourceObjectAttributes;
ServerBISSubscriptionCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "channels",
        "baseName": "channels",
        "type": "Array<ServerBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum>"
    },
    {
        "name": "profile",
        "baseName": "profile",
        "type": "ServerBISSubscriptionCreateQueryResourceObjectAttributesProfile"
    }
];
(function (ServerBISSubscriptionCreateQueryResourceObjectAttributes) {
    let ChannelsEnum;
    (function (ChannelsEnum) {
        ChannelsEnum[ChannelsEnum["Email"] = 'EMAIL'] = "Email";
        ChannelsEnum[ChannelsEnum["Push"] = 'PUSH'] = "Push";
        ChannelsEnum[ChannelsEnum["Sms"] = 'SMS'] = "Sms";
    })(ChannelsEnum = ServerBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum || (ServerBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum = {}));
})(ServerBISSubscriptionCreateQueryResourceObjectAttributes = exports.ServerBISSubscriptionCreateQueryResourceObjectAttributes || (exports.ServerBISSubscriptionCreateQueryResourceObjectAttributes = {}));
