// klaviyo-api/dist/model/staticDateFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticDateFilter = void 0;
class StaticDateFilter {
    static getAttributeTypeMap() {
        return StaticDateFilter.attributeTypeMap;
    }
}
exports.StaticDateFilter = StaticDateFilter;
StaticDateFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "DateEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "StaticDateFilter.OperatorEnum"
    },
    {
        "name": "date",
        "baseName": "date",
        "type": "Date"
    }
];
(function (StaticDateFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["After"] = 'after'] = "After";
        OperatorEnum[OperatorEnum["Before"] = 'before'] = "Before";
    })(OperatorEnum = StaticDateFilter.OperatorEnum || (StaticDateFilter.OperatorEnum = {}));
})(StaticDateFilter = exports.StaticDateFilter || (exports.StaticDateFilter = {}));
