// klaviyo-api/dist/model/stringOperatorFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringOperatorFilter = void 0;
class StringOperatorFilter {
    static getAttributeTypeMap() {
        return StringOperatorFilter.attributeTypeMap;
    }
}
exports.StringOperatorFilter = StringOperatorFilter;
StringOperatorFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "StringEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "StringOperatorFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (StringOperatorFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["Contains"] = 'contains'] = "Contains";
        OperatorEnum[OperatorEnum["EndsWith"] = 'ends-with'] = "EndsWith";
        OperatorEnum[OperatorEnum["Equals"] = 'equals'] = "Equals";
        OperatorEnum[OperatorEnum["NotContains"] = 'not-contains'] = "NotContains";
        OperatorEnum[OperatorEnum["NotEndsWith"] = 'not-ends-with'] = "NotEndsWith";
        OperatorEnum[OperatorEnum["NotEquals"] = 'not-equals'] = "NotEquals";
        OperatorEnum[OperatorEnum["NotStartsWith"] = 'not-starts-with'] = "NotStartsWith";
        OperatorEnum[OperatorEnum["StartsWith"] = 'starts-with'] = "StartsWith";
    })(OperatorEnum = StringOperatorFilter.OperatorEnum || (StringOperatorFilter.OperatorEnum = {}));
})(StringOperatorFilter = exports.StringOperatorFilter || (exports.StringOperatorFilter = {}));
