// klaviyo-api/dist/model/stringArrayOperatorFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringArrayOperatorFilter = void 0;
class StringArrayOperatorFilter {
    static getAttributeTypeMap() {
        return StringArrayOperatorFilter.attributeTypeMap;
    }
}
exports.StringArrayOperatorFilter = StringArrayOperatorFilter;
StringArrayOperatorFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "StringEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "StringArrayOperatorFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<string>"
    }
];
(function (StringArrayOperatorFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["In"] = 'in'] = "In";
        OperatorEnum[OperatorEnum["NotIn"] = 'not-in'] = "NotIn";
    })(OperatorEnum = StringArrayOperatorFilter.OperatorEnum || (StringArrayOperatorFilter.OperatorEnum = {}));
})(StringArrayOperatorFilter = exports.StringArrayOperatorFilter || (exports.StringArrayOperatorFilter = {}));
