// klaviyo-api/dist/model/stringPhoneOperatorArrayFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringPhoneOperatorArrayFilter = void 0;
class StringPhoneOperatorArrayFilter {
    static getAttributeTypeMap() {
        return StringPhoneOperatorArrayFilter.attributeTypeMap;
    }
}
exports.StringPhoneOperatorArrayFilter = StringPhoneOperatorArrayFilter;
StringPhoneOperatorArrayFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "StringEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "StringPhoneOperatorArrayFilter.OperatorEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<string>"
    }
];
(function (StringPhoneOperatorArrayFilter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["In"] = 'phone-country-code-in'] = "In";
        OperatorEnum[OperatorEnum["NotIn"] = 'phone-country-code-not-in'] = "NotIn";
    })(OperatorEnum = StringPhoneOperatorArrayFilter.OperatorEnum || (StringPhoneOperatorArrayFilter.OperatorEnum = {}));
})(StringPhoneOperatorArrayFilter = exports.StringPhoneOperatorArrayFilter || (exports.StringPhoneOperatorArrayFilter = {}));
