// klaviyo-api/dist/model/throttledSendStrategy.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThrottledSendStrategy = void 0;
class ThrottledSendStrategy {
    static getAttributeTypeMap() {
        return ThrottledSendStrategy.attributeTypeMap;
    }
}
exports.ThrottledSendStrategy = ThrottledSendStrategy;
ThrottledSendStrategy.attributeTypeMap = [
    {
        "name": "method",
        "baseName": "method",
        "type": "ThrottledEnum"
    },
    {
        "name": "datetime",
        "baseName": "datetime",
        "type": "Date"
    },
    {
        "name": "throttlePercentage",
        "baseName": "throttle_percentage",
        "type": "ThrottledSendStrategy.ThrottlePercentageEnum"
    }
];
(function (ThrottledSendStrategy) {
    let ThrottlePercentageEnum;
    (function (ThrottlePercentageEnum) {
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_10"] = 10] = "NUMBER_10";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_11"] = 11] = "NUMBER_11";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_13"] = 13] = "NUMBER_13";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_14"] = 14] = "NUMBER_14";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_17"] = 17] = "NUMBER_17";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_20"] = 20] = "NUMBER_20";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_25"] = 25] = "NUMBER_25";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_33"] = 33] = "NUMBER_33";
        ThrottlePercentageEnum[ThrottlePercentageEnum["NUMBER_50"] = 50] = "NUMBER_50";
    })(ThrottlePercentageEnum = ThrottledSendStrategy.ThrottlePercentageEnum || (ThrottledSendStrategy.ThrottlePercentageEnum = {}));
})(ThrottledSendStrategy = exports.ThrottledSendStrategy || (exports.ThrottledSendStrategy = {}));
