// klaviyo-api/dist/model/timeframe.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Timeframe = void 0;
class Timeframe {
    static getAttributeTypeMap() {
        return Timeframe.attributeTypeMap;
    }
}
exports.Timeframe = Timeframe;
Timeframe.attributeTypeMap = [
    {
        "name": "key",
        "baseName": "key",
        "type": "Timeframe.KeyEnum"
    }
];
(function (Timeframe) {
    let KeyEnum;
    (function (KeyEnum) {
        KeyEnum[KeyEnum["Last12Months"] = 'last_12_months'] = "Last12Months";
        KeyEnum[KeyEnum["Last30Days"] = 'last_30_days'] = "Last30Days";
        KeyEnum[KeyEnum["Last365Days"] = 'last_365_days'] = "Last365Days";
        KeyEnum[KeyEnum["Last3Months"] = 'last_3_months'] = "Last3Months";
        KeyEnum[KeyEnum["Last7Days"] = 'last_7_days'] = "Last7Days";
        KeyEnum[KeyEnum["Last90Days"] = 'last_90_days'] = "Last90Days";
        KeyEnum[KeyEnum["LastMonth"] = 'last_month'] = "LastMonth";
        KeyEnum[KeyEnum["LastWeek"] = 'last_week'] = "LastWeek";
        KeyEnum[KeyEnum["LastYear"] = 'last_year'] = "LastYear";
        KeyEnum[KeyEnum["ThisMonth"] = 'this_month'] = "ThisMonth";
        KeyEnum[KeyEnum["ThisWeek"] = 'this_week'] = "ThisWeek";
        KeyEnum[KeyEnum["ThisYear"] = 'this_year'] = "ThisYear";
        KeyEnum[KeyEnum["Today"] = 'today'] = "Today";
        KeyEnum[KeyEnum["Yesterday"] = 'yesterday'] = "Yesterday";
    })(KeyEnum = Timeframe.KeyEnum || (Timeframe.KeyEnum = {}));
})(Timeframe = exports.Timeframe || (exports.Timeframe = {}));
