// klaviyo-api/dist/model/triggerBranchActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerBranchActionData = void 0;
class TriggerBranchActionData {
    static getAttributeTypeMap() {
        return TriggerBranchActionData.attributeTypeMap;
    }
}
exports.TriggerBranchActionData = TriggerBranchActionData;
TriggerBranchActionData.attributeTypeMap = [
    {
        "name": "triggerFilter",
        "baseName": "trigger_filter",
        "type": "TriggerBranchActionDataTriggerFilter"
    },
    {
        "name": "triggerId",
        "baseName": "trigger_id",
        "type": "string"
    },
    {
        "name": "triggerType",
        "baseName": "trigger_type",
        "type": "TriggerBranchActionData.TriggerTypeEnum"
    },
    {
        "name": "triggerSubtype",
        "baseName": "trigger_subtype",
        "type": "TriggerBranchActionData.TriggerSubtypeEnum"
    }
];
(function (TriggerBranchActionData) {
    let TriggerTypeEnum;
    (function (TriggerTypeEnum) {
        TriggerTypeEnum[TriggerTypeEnum["Date"] = 'date'] = "Date";
        TriggerTypeEnum[TriggerTypeEnum["List"] = 'list'] = "List";
        TriggerTypeEnum[TriggerTypeEnum["LowInventory"] = 'low-inventory'] = "LowInventory";
        TriggerTypeEnum[TriggerTypeEnum["Metric"] = 'metric'] = "Metric";
        TriggerTypeEnum[TriggerTypeEnum["PriceDrop"] = 'price-drop'] = "PriceDrop";
        TriggerTypeEnum[TriggerTypeEnum["Scheduled"] = 'scheduled'] = "Scheduled";
        TriggerTypeEnum[TriggerTypeEnum["Segment"] = 'segment'] = "Segment";
    })(TriggerTypeEnum = TriggerBranchActionData.TriggerTypeEnum || (TriggerBranchActionData.TriggerTypeEnum = {}));
    let TriggerSubtypeEnum;
    (function (TriggerSubtypeEnum) {
        TriggerSubtypeEnum[TriggerSubtypeEnum["CustomObject"] = 'custom-object'] = "CustomObject";
        TriggerSubtypeEnum[TriggerSubtypeEnum["LegacyCustomObject"] = 'legacy-custom-object'] = "LegacyCustomObject";
        TriggerSubtypeEnum[TriggerSubtypeEnum["ProfileProperty"] = 'profile-property'] = "ProfileProperty";
        TriggerSubtypeEnum[TriggerSubtypeEnum["ProfileTrait"] = 'profile-trait'] = "ProfileTrait";
    })(TriggerSubtypeEnum = TriggerBranchActionData.TriggerSubtypeEnum || (TriggerBranchActionData.TriggerSubtypeEnum = {}));
})(TriggerBranchActionData = exports.TriggerBranchActionData || (exports.TriggerBranchActionData = {}));
