// klaviyo-api/dist/model/universalContentResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniversalContentResponseObjectResourceAttributes = void 0;
class UniversalContentResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return UniversalContentResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.UniversalContentResponseObjectResourceAttributes = UniversalContentResponseObjectResourceAttributes;
UniversalContentResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "definition",
        "baseName": "definition",
        "type": "UniversalContentResponseObjectResourceAttributesDefinition"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "screenshotStatus",
        "baseName": "screenshot_status",
        "type": "UniversalContentResponseObjectResourceAttributes.ScreenshotStatusEnum"
    },
    {
        "name": "screenshotUrl",
        "baseName": "screenshot_url",
        "type": "string"
    }
];
(function (UniversalContentResponseObjectResourceAttributes) {
    let ScreenshotStatusEnum;
    (function (ScreenshotStatusEnum) {
        ScreenshotStatusEnum[ScreenshotStatusEnum["Completed"] = 'completed'] = "Completed";
        ScreenshotStatusEnum[ScreenshotStatusEnum["Failed"] = 'failed'] = "Failed";
        ScreenshotStatusEnum[ScreenshotStatusEnum["Generating"] = 'generating'] = "Generating";
        ScreenshotStatusEnum[ScreenshotStatusEnum["NeverGenerated"] = 'never_generated'] = "NeverGenerated";
        ScreenshotStatusEnum[ScreenshotStatusEnum["NotRenderable"] = 'not_renderable'] = "NotRenderable";
        ScreenshotStatusEnum[ScreenshotStatusEnum["Stale"] = 'stale'] = "Stale";
    })(ScreenshotStatusEnum = UniversalContentResponseObjectResourceAttributes.ScreenshotStatusEnum || (UniversalContentResponseObjectResourceAttributes.ScreenshotStatusEnum = {}));
})(UniversalContentResponseObjectResourceAttributes = exports.UniversalContentResponseObjectResourceAttributes || (exports.UniversalContentResponseObjectResourceAttributes = {}));
