// klaviyo-api/dist/model/updateProfileActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateProfileActionData = void 0;
class UpdateProfileActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = UpdateProfileActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return UpdateProfileActionData.attributeTypeMap;
    }
}
exports.UpdateProfileActionData = UpdateProfileActionData;
UpdateProfileActionData.attributeTypeMap = [
    {
        "name": "profileOperations",
        "baseName": "profile_operations",
        "type": "Array<UpdateProfileActionDataProfileOperationsInner>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "UpdateProfileActionData.StatusEnum"
    }
];
(function (UpdateProfileActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = UpdateProfileActionData.StatusEnum || (UpdateProfileActionData.StatusEnum = {}));
})(UpdateProfileActionData = exports.UpdateProfileActionData || (exports.UpdateProfileActionData = {}));
