// klaviyo-api/dist/model/webFeedResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebFeedResponseObjectResourceAttributes = void 0;
class WebFeedResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return WebFeedResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.WebFeedResponseObjectResourceAttributes = WebFeedResponseObjectResourceAttributes;
WebFeedResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "requestMethod",
        "baseName": "request_method",
        "type": "WebFeedResponseObjectResourceAttributes.RequestMethodEnum"
    },
    {
        "name": "contentType",
        "baseName": "content_type",
        "type": "WebFeedResponseObjectResourceAttributes.ContentTypeEnum"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "WebFeedResponseObjectResourceAttributes.StatusEnum"
    }
];
(function (WebFeedResponseObjectResourceAttributes) {
    let RequestMethodEnum;
    (function (RequestMethodEnum) {
        RequestMethodEnum[RequestMethodEnum["Get"] = 'get'] = "Get";
        RequestMethodEnum[RequestMethodEnum["Post"] = 'post'] = "Post";
    })(RequestMethodEnum = WebFeedResponseObjectResourceAttributes.RequestMethodEnum || (WebFeedResponseObjectResourceAttributes.RequestMethodEnum = {}));
    let ContentTypeEnum;
    (function (ContentTypeEnum) {
        ContentTypeEnum[ContentTypeEnum["Json"] = 'json'] = "Json";
        ContentTypeEnum[ContentTypeEnum["Xml"] = 'xml'] = "Xml";
    })(ContentTypeEnum = WebFeedResponseObjectResourceAttributes.ContentTypeEnum || (WebFeedResponseObjectResourceAttributes.ContentTypeEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["CriticalNightlyRefreshTimeout"] = 'critical_nightly_refresh_timeout'] = "CriticalNightlyRefreshTimeout";
        StatusEnum[StatusEnum["Disabled"] = 'disabled'] = "Disabled";
        StatusEnum[StatusEnum["Ok"] = 'ok'] = "Ok";
        StatusEnum[StatusEnum["WarningNightlyRefreshTimeout"] = 'warning_nightly_refresh_timeout'] = "WarningNightlyRefreshTimeout";
        StatusEnum[StatusEnum["WarningPeriodicRefreshTimeout"] = 'warning_periodic_refresh_timeout'] = "WarningPeriodicRefreshTimeout";
    })(StatusEnum = WebFeedResponseObjectResourceAttributes.StatusEnum || (WebFeedResponseObjectResourceAttributes.StatusEnum = {}));
})(WebFeedResponseObjectResourceAttributes = exports.WebFeedResponseObjectResourceAttributes || (exports.WebFeedResponseObjectResourceAttributes = {}));
