// klaviyo-api/dist/model/webFeedCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebFeedCreateQueryResourceObjectAttributes = void 0;
class WebFeedCreateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return WebFeedCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.WebFeedCreateQueryResourceObjectAttributes = WebFeedCreateQueryResourceObjectAttributes;
WebFeedCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "requestMethod",
        "baseName": "request_method",
        "type": "WebFeedCreateQueryResourceObjectAttributes.RequestMethodEnum"
    },
    {
        "name": "contentType",
        "baseName": "content_type",
        "type": "WebFeedCreateQueryResourceObjectAttributes.ContentTypeEnum"
    }
];
(function (WebFeedCreateQueryResourceObjectAttributes) {
    let RequestMethodEnum;
    (function (RequestMethodEnum) {
        RequestMethodEnum[RequestMethodEnum["Get"] = 'get'] = "Get";
        RequestMethodEnum[RequestMethodEnum["Post"] = 'post'] = "Post";
    })(RequestMethodEnum = WebFeedCreateQueryResourceObjectAttributes.RequestMethodEnum || (WebFeedCreateQueryResourceObjectAttributes.RequestMethodEnum = {}));
    let ContentTypeEnum;
    (function (ContentTypeEnum) {
        ContentTypeEnum[ContentTypeEnum["Json"] = 'json'] = "Json";
        ContentTypeEnum[ContentTypeEnum["Xml"] = 'xml'] = "Xml";
    })(ContentTypeEnum = WebFeedCreateQueryResourceObjectAttributes.ContentTypeEnum || (WebFeedCreateQueryResourceObjectAttributes.ContentTypeEnum = {}));
})(WebFeedCreateQueryResourceObjectAttributes = exports.WebFeedCreateQueryResourceObjectAttributes || (exports.WebFeedCreateQueryResourceObjectAttributes = {}));
