// klaviyo-api/dist/model/webFeedPartialUpdateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebFeedPartialUpdateQueryResourceObjectAttributes = void 0;
class WebFeedPartialUpdateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return WebFeedPartialUpdateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.WebFeedPartialUpdateQueryResourceObjectAttributes = WebFeedPartialUpdateQueryResourceObjectAttributes;
WebFeedPartialUpdateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "requestMethod",
        "baseName": "request_method",
        "type": "WebFeedPartialUpdateQueryResourceObjectAttributes.RequestMethodEnum"
    },
    {
        "name": "contentType",
        "baseName": "content_type",
        "type": "WebFeedPartialUpdateQueryResourceObjectAttributes.ContentTypeEnum"
    }
];
(function (WebFeedPartialUpdateQueryResourceObjectAttributes) {
    let RequestMethodEnum;
    (function (RequestMethodEnum) {
        RequestMethodEnum[RequestMethodEnum["Get"] = 'get'] = "Get";
        RequestMethodEnum[RequestMethodEnum["Post"] = 'post'] = "Post";
    })(RequestMethodEnum = WebFeedPartialUpdateQueryResourceObjectAttributes.RequestMethodEnum || (WebFeedPartialUpdateQueryResourceObjectAttributes.RequestMethodEnum = {}));
    let ContentTypeEnum;
    (function (ContentTypeEnum) {
        ContentTypeEnum[ContentTypeEnum["Json"] = 'json'] = "Json";
        ContentTypeEnum[ContentTypeEnum["Xml"] = 'xml'] = "Xml";
    })(ContentTypeEnum = WebFeedPartialUpdateQueryResourceObjectAttributes.ContentTypeEnum || (WebFeedPartialUpdateQueryResourceObjectAttributes.ContentTypeEnum = {}));
})(WebFeedPartialUpdateQueryResourceObjectAttributes = exports.WebFeedPartialUpdateQueryResourceObjectAttributes || (exports.WebFeedPartialUpdateQueryResourceObjectAttributes = {}));
